Disabling depth makes zed_wrapper crash. Position tracking cannot be disabled and runs every time.

### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues.
- [X] This issue is not a question, feature request, or anything other than a bug report directly related to this project.

### Description

I have noticed, that whenever I try to disable the depth map computation "commo.yaml->depth:quality", the zed_node crashes without giving any appropriate error message. 
Also it seems that you cannot disable position tracking with the "common.yaml->pos_tracking_enabled" flag. 

Maybe I have misunderstood something in your documentation, there might be some setting in conflict causing all these problems and in that case I sincerely apologize for my incompetence and blindness.

Here is the my "common.yaml" file:

        general:
            svo_file:                   ''
            svo_loop:                   false                               # Enable loop mode when using an SVO as input source
            svo_realtime:               false                               
            camera_timeout_sec:         5
            camera_max_reconnect:       5
            camera_flip:                false
            zed_id:                     0
            serial_number:              0
            resolution:                 3                                   # '0': HD2K, '1': HD1080, '2': HD720, '3': VGA
            sdk_verbose:                true
            grab_frame_rate:            60                                # ZED SDK internal grabbing rate
            pub_frame_rate:             40.0                                # [DYNAMIC] - frequency of publishing of visual images and depth images
            gpu_id:                     -1

        video:
            extrinsic_in_camera_frame:  false                               # if `false` extrinsic parameter in `camera_info` will use ROS native frame (X FORWARD, Z UP) instead of the camera frame (Z FORWARD, Y DOWN) [`true` use old behavior as for version < v3.1]
            img_downsample_factor:      1.0                                 # Resample factor for image data matrices [0.01,1.0] The SDK works with native data sizes, but publishes rescaled matrices
            brightness:                 4                                   # [DYNAMIC]
            contrast:                   4                                   # [DYNAMIC]
            hue:                        0                                   # [DYNAMIC]
            saturation:                 4                                   # [DYNAMIC]
            sharpness:                  4                                   # [DYNAMIC]
            gamma:                      8                                   # [DYNAMIC] - Requires SDK >=v3.1
            auto_exposure_gain:         true                                # [DYNAMIC]
            exposure:                   80                                  # [DYNAMIC]
            gain:                       80                                  # [DYNAMIC]
            auto_whitebalance:          true                                # [DYNAMIC]
            whitebalance_temperature:   42                                  # [DYNAMIC] - [28,65] works only if `auto_whitebalance` is false
            qos_history:                1                                   # '1': KEEP_LAST - '2': KEEP_ALL
            qos_depth:                  1                                   # Queue size if using KEEP_LAST
            qos_reliability:            1                                   # '1': RELIABLE - '2': BEST_EFFORT -
            qos_durability:             1                                   # '1': TRANSIENT_LOCAL - '2': VOLATILE

        depth:
            quality:                    0                                   # '0': NONE, '1': PERFORMANCE, '2': QUALITY, '3': ULTRA - Note: if '0' all the modules that requires depth extraction are disabled by default (Pos. Tracking, Obj. Detection, Mapping, ...)
            sensing_mode:               0                                   # '0': STANDARD, '1': FILL
            depth_stabilization:        false                                # Forces positional tracking to start if 'true'
            openni_depth_mode:          false                               # 'false': 32bit float [meters], 'true': 16bit unsigned int [millimeters]
            depth_downsample_factor:    0.1                                 # Resample factor for depth data matrices [0.01,1.0] The SDK works with native data sizes, but publishes rescaled matrices (depth map, point cloud, ...)
            point_cloud_freq:           0.1                                # [DYNAMIC] - frequency of the pointcloud publishing (equal or less to `grab_frame_rate` value)
            depth_confidence:           50                                  # [DYNAMIC]
            depth_texture_conf:         100                                 # [DYNAMIC]
            remove_saturated_areas:     true                                # [DYNAMIC]
            qos_history:                1                                   # '1': KEEP_LAST - '2': KEEP_ALL
            qos_depth:                  1                                   # Queue size if using KEEP_LAST
            qos_reliability:            1                                   # '1': RELIABLE - '2': BEST_EFFORT -
            qos_durability:             2                                   # '1': TRANSIENT_LOCAL - '2': VOLATILE

        pos_tracking:
            pos_tracking_enabled:       true                               # True to enable positional tracking from start
            publish_tf:                 true                                # publish `odom -> base_link` TF
            publish_map_tf:             true                                # publish `map -> odom` TF
            base_frame:                 'base_link'                         # use the same name as in the URDF file
            map_frame:                  'map'
            odometry_frame:             'vis_odom'
            area_memory_db_path:        ''
            area_memory:                false                                # Enable to detect loop closure
            floor_alignment:            false                                # Enable to automatically calculate camera/floor offset
            initial_base_pose:          [0.0,0.0,0.0, 0.0,0.0,0.0]          # Initial position of the `base_frame` -> [X, Y, Z, R, P, Y]
            init_odom_with_first_valid_pose: true                           # Enable to initialize the odometry with the first valid pose
            path_pub_rate:              10.0                                # [DYNAMIC] - Camera trajectory publishing frequency
            path_max_count:             10                                 # use '-1' for unlimited path size
            two_d_mode:                 false                               # Force navigation on a plane. If true the Z value will be fixed to "fixed_z_value", roll and pitch to zero
            fixed_z_value:              0.00                                # Value to be used for Z coordinate if `two_d_mode` is true
            qos_history:                1                                   # '1': KEEP_LAST - '2': KEEP_ALL
            qos_depth:                  1                                   # Queue size if using KEEP_LAST
            qos_reliability:            1                                   # '1': RELIABLE - '2': BEST_EFFORT -
            qos_durability:             1                                   # '1': TRANSIENT_LOCAL - '2': VOLATILE

        mapping:
            mapping_enabled:            false                               # True to enable mapping and fused point cloud pubblication
            resolution:                 0.1                                # maps resolution in meters [0.01f, 0.2f]
            max_mapping_range:          20.0                                  # maximum depth range while mapping in meters (-1 for automatic calculation) [2.0, 20.0]
            fused_pointcloud_freq:      0.1                                 # frequency of the publishing of the fused colored point cloud
            qos_history:                1                                   # '1': KEEP_LAST - '2': KEEP_ALL
            qos_depth:                  1                                   # Queue size if using KEEP_LAST
            qos_reliability:            1                                   # '1': RELIABLE - '2': BEST_EFFORT -
            qos_durability:             1                                   # '1': TRANSIENT_LOCAL - '2': VOLATILE

        debug:
            debug_general:              false
            debug_sensors:              false


and  here is "zed2.yaml":

        general:
            camera_model:               'zed2'
            camera_name:                'zed2'

        depth:
            min_depth:                  0.7            # Min: 0.2, Max: 3.0 - Default 0.7 - Note: reducing this value wil require more computational power and GPU memory
            max_depth:                  20.0            # Max: 40.0

        pos_tracking:
            imu_fusion:                 true            # enable/disable IMU fusion. When set to false, only the optical odometry will be used.
            publish_imu_tf:             true            # enable/disable the static IMU TF broadcasting

        sensors:
            sensors_image_sync:         false           # Synchronize Sensors messages with latest published video/depth message
            sensors_pub_rate:           200.            # frequency of publishing of sensors data. MAX: 400. - MIN: grab rate
            qos_history:                1               # '1': KEEP_LAST - '2': KEEP_ALL
            qos_depth:                  10               # Queue size if using KEEP_LAST
            qos_reliability:            1               # '1': RELIABLE - '2': BEST_EFFORT -
            qos_durability:             2               # '1': TRANSIENT_LOCAL - '2': VOLATILE

        object_detection:
            od_enabled:                 false           # True to enable Object Detection [only ZED 2]
            confidence_threshold:       50.0            # [DYNAMIC] - Minimum value of the detection confidence of an object [0,100]
            model:                      0               # '0': MULTI_CLASS_BOX - '1': MULTI_CLASS_BOX_ACCURATE - '2': HUMAN_BODY_FAST - '3': HUMAN_BODY_ACCURATE - '4': MULTI_CLASS_BOX_MEDIUM - '5': HUMAN_BODY_MEDIUM - '6': PERSON_HEAD_BOX 
            filtering_mode:             1               # '0': NONE - '1': NMS3D - '2': NMS3D_PER_CLASS
            mc_people:                  true            # [DYNAMIC] - Enable/disable the detection of persons for 'MULTI_CLASS_X' models
            mc_vehicle:                 true            # [DYNAMIC] - Enable/disable the detection of vehicles for 'MULTI_CLASS_X' models
            mc_bag:                     true            # [DYNAMIC] - Enable/disable the detection of bags for 'MULTI_CLASS_X' models
            mc_animal:                  true            # [DYNAMIC] - Enable/disable the detection of animals for 'MULTI_CLASS_X' models
            mc_electronics:             true            # [DYNAMIC] - Enable/disable the detection of electronic devices for 'MULTI_CLASS_X' models
            mc_fruit_vegetable:         true            # [DYNAMIC] - Enable/disable the detection of fruits and vegetables for 'MULTI_CLASS_X' models
            mc_sport:                   true            # [DYNAMIC] - Enable/disable the detection of sport-related objects for 'MULTI_CLASS_X' models
            body_fitting:               false           # Enable/disable body fitting for 'HUMAN_BODY_FAST' and 'HUMAN_BODY_ACCURATE' models
            body_format:                1               # '0': POSE_18 - '1': POSE_34 [Only if `HUMAN_BODY_*` model is selected]
            qos_history:                1               # '1': KEEP_LAST - '2': KEEP_ALL
            qos_depth:                  1               # Queue size if using KEEP_LAST
            qos_reliability:            2               # '1': RELIABLE - '2': BEST_EFFORT
            qos_durability:             2               # '1': TRANSIENT_LOCAL - '2': VOLATILE

### Steps to Reproduce

1. run docker image
2. launch zed wrapper with "ros2 launch zed_wrapper zed2.launch.py"

Steps to reproduce are the same for both instances, however with some differences in the "common.yaml".
Depth crashes with "depth:quality:0" setting, and position tracking is running even though "pos_tracking:pos_tracking_enabled:0" is set. 



### Expected Result

I don't really know how would the correct console output look like for either one of these options.

### Actual Result

### For the setting with depth turned off:

ros2 launch zed_wrapper zed2.launch.py 
[INFO] [launch]: All log files can be found below /root/.ros/log/2022-06-08-16-07-55-121876-ubuntu-5393
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [robot_state_publisher-1]: process started with pid [5399]
[INFO] [zed_wrapper-2]: process started with pid [5400]
[robot_state_publisher-1] Parsing robot urdf xml string.
[robot_state_publisher-1] Link zed2_base_link had 1 children
[robot_state_publisher-1] Link zed2_camera_center had 4 children
[robot_state_publisher-1] Link zed2_baro_link had 0 children
[robot_state_publisher-1] Link zed2_left_camera_frame had 2 children
[robot_state_publisher-1] Link zed2_left_camera_optical_frame had 0 children
[robot_state_publisher-1] Link zed2_temp_left_link had 0 children
[robot_state_publisher-1] Link zed2_mag_link had 0 children
[robot_state_publisher-1] Link zed2_right_camera_frame had 2 children
[robot_state_publisher-1] Link zed2_right_camera_optical_frame had 0 children
[robot_state_publisher-1] Link zed2_temp_right_link had 0 children
[robot_state_publisher-1] 1654704475.615523391 [zed2.zed_state_publisher] [INFO] got segment base_link
[robot_state_publisher-1] 1654704475.615771529 [zed2.zed_state_publisher] [INFO] got segment zed2_baro_link
[robot_state_publisher-1] 1654704475.615827724 [zed2.zed_state_publisher] [INFO] got segment zed2_base_link
[robot_state_publisher-1] 1654704475.615870734 [zed2.zed_state_publisher] [INFO] got segment zed2_camera_center
[robot_state_publisher-1] 1654704475.615905903 [zed2.zed_state_publisher] [INFO] got segment zed2_left_camera_frame
[robot_state_publisher-1] 1654704475.615951217 [zed2.zed_state_publisher] [INFO] got segment zed2_left_camera_optical_frame
[robot_state_publisher-1] 1654704475.615976242 [zed2.zed_state_publisher] [INFO] got segment zed2_mag_link
[robot_state_publisher-1] 1654704475.615999763 [zed2.zed_state_publisher] [INFO] got segment zed2_right_camera_frame
[robot_state_publisher-1] 1654704475.616020916 [zed2.zed_state_publisher] [INFO] got segment zed2_right_camera_optical_frame
[robot_state_publisher-1] 1654704475.616041941 [zed2.zed_state_publisher] [INFO] got segment zed2_temp_left_link
[robot_state_publisher-1] 1654704475.616063190 [zed2.zed_state_publisher] [INFO] got segment zed2_temp_right_link
[zed_wrapper-2] 1654704475.681843919 [zed2.zed_node] [INFO] ********************************
[zed_wrapper-2] 1654704475.682205470 [zed2.zed_node] [INFO]       ZED Camera Component 
[zed_wrapper-2] 1654704475.682248256 [zed2.zed_node] [INFO] ********************************
[zed_wrapper-2] 1654704475.682315235 [zed2.zed_node] [INFO]  * namespace: /zed2
[zed_wrapper-2] 1654704475.682338980 [zed2.zed_node] [INFO]  * node name: zed_node
[zed_wrapper-2] 1654704475.682359685 [zed2.zed_node] [INFO] ********************************
[zed_wrapper-2] 1654704475.682395046 [zed2.zed_node] [INFO] *** DEBUG parameters ***
[zed_wrapper-2] 1654704475.682545357 [zed2.zed_node] [INFO]  * Debug mode: FALSE
[zed_wrapper-2] 1654704475.682591983 [zed2.zed_node] [INFO] *** GENERAL parameters ***
[zed_wrapper-2] 1654704475.682684883 [zed2.zed_node] [INFO]  * Camera model: zed2 - ZED 2
[zed_wrapper-2] 1654704475.682766102 [zed2.zed_node] [INFO]  * SDK Verbose: 1
[zed_wrapper-2] 1654704475.682846618 [zed2.zed_node] [INFO]  * SVO: live
[zed_wrapper-2] 1654704475.682950974 [zed2.zed_node] [INFO]  * SVO Loop: FALSE
[zed_wrapper-2] 1654704475.683007744 [zed2.zed_node] [INFO]  * SVO Realtime: FALSE
[zed_wrapper-2] 1654704475.683065507 [zed2.zed_node] [INFO]  * Camera name: zed2
[zed_wrapper-2] 1654704475.683209257 [zed2.zed_node] [INFO]  * Camera ID: 0
[zed_wrapper-2] 1654704475.683271020 [zed2.zed_node] [INFO]  * Camera SN: 0
[zed_wrapper-2] 1654704475.683338991 [zed2.zed_node] [INFO]  * Camera timeout [sec]: 5
[zed_wrapper-2] 1654704475.683395441 [zed2.zed_node] [INFO]  * Camera reconnection temptatives: 5
[zed_wrapper-2] 1654704475.683496053 [zed2.zed_node] [INFO]  * Camera framerate: 60
[zed_wrapper-2] 1654704475.683573273 [zed2.zed_node] [INFO]  * GPU ID: -1
[zed_wrapper-2] 1654704475.683652252 [zed2.zed_node] [INFO]  * Camera resolution: 3 - VGA
[zed_wrapper-2] 1654704475.683711455 [zed2.zed_node] [INFO]  * Camera self calibration: TRUE
[zed_wrapper-2] 1654704475.683769761 [zed2.zed_node] [INFO]  * Camera flip: FALSE
[zed_wrapper-2] 1654704475.683823939 [zed2.zed_node] [INFO]  * [DYN] Publish framerate [Hz]: 40 
[zed_wrapper-2] 1654704475.683858533 [zed2.zed_node] [INFO] *** VIDEO parameters ***
[zed_wrapper-2] 1654704475.683916167 [zed2.zed_node] [INFO]  * Use old extrinsic parameters: 0
[zed_wrapper-2] 1654704475.683988938 [zed2.zed_node] [INFO]  * [DYN] Image downsample factor: 1 
[zed_wrapper-2] 1654704475.684087503 [zed2.zed_node] [INFO]  * [DYN] Brightness: 4
[zed_wrapper-2] 1654704475.684144561 [zed2.zed_node] [INFO]  * [DYN] Contrast: 4
[zed_wrapper-2] 1654704475.684220564 [zed2.zed_node] [INFO]  * [DYN] Hue: 0
[zed_wrapper-2] 1654704475.684275543 [zed2.zed_node] [INFO]  * [DYN] Saturation: 4
[zed_wrapper-2] 1654704475.684337049 [zed2.zed_node] [INFO]  * [DYN] Sharpness: 4
[zed_wrapper-2] 1654704475.684390172 [zed2.zed_node] [INFO]  * [DYN] Gamma: 8
[zed_wrapper-2] 1654704475.684440158 [zed2.zed_node] [INFO]  * [DYN] Auto Exposure/Gain: TRUE
[zed_wrapper-2] 1654704475.684490400 [zed2.zed_node] [INFO]  * [DYN] Exposure: 80
[zed_wrapper-2] 1654704475.684541506 [zed2.zed_node] [INFO]  * [DYN] Gain: 80
[zed_wrapper-2] 1654704475.684603013 [zed2.zed_node] [INFO]  * [DYN] Auto White Balance: TRUE
[zed_wrapper-2] 1654704475.684654951 [zed2.zed_node] [INFO]  * [DYN] White Balance Temperature: 42
[zed_wrapper-2] 1654704475.684709609 [zed2.zed_node] [INFO]  * Video QoS History: KEEP_LAST
[zed_wrapper-2] 1654704475.684757099 [zed2.zed_node] [INFO]  * Video QoS History depth: 1
[zed_wrapper-2] 1654704475.685904060 [zed2.zed_node] [INFO]  * Video QoS Reliability: RELIABLE
[zed_wrapper-2] 1654704475.686051106 [zed2.zed_node] [INFO]  * Video QoS Durability: VOLATILE
[zed_wrapper-2] 1654704475.686086084 [zed2.zed_node] [INFO] *** DEPTH parameters ***
[zed_wrapper-2] 1654704475.686152295 [zed2.zed_node] [INFO]  * Depth quality: 0 - NONE - DEPTH DISABLED
[zed_wrapper-2] 1654704475.686206953 [zed2.zed_node] [INFO] *** POSITIONAL TRACKING parameters ***
[zed_wrapper-2] 1654704475.686265932 [zed2.zed_node] [INFO]  * Positional tracking enabled: TRUE
[zed_wrapper-2] 1654704475.686342415 [zed2.zed_node] [INFO]  * Base frame id: base_link
[zed_wrapper-2] 1654704475.686403473 [zed2.zed_node] [INFO]  * Map frame id: map
[zed_wrapper-2] 1654704475.686605146 [zed2.zed_node] [INFO]  * Odometry frame id: vis_odom
[zed_wrapper-2] 1654704475.686661308 [zed2.zed_node] [INFO]  * Broadcast Odometry TF: TRUE
[zed_wrapper-2] 1654704475.686721087 [zed2.zed_node] [INFO]  * Broadcast Pose TF: TRUE
[zed_wrapper-2] 1654704475.686788098 [zed2.zed_node] [INFO]  * Broadcast Static IMU TF [not for ZED]: TRUE
[zed_wrapper-2] 1654704475.686923304 [zed2.zed_node] [INFO]  * [DYN] Path publishing rate: 10
[zed_wrapper-2] 1654704475.686988778 [zed2.zed_node] [INFO]  * Path history lenght: 10
[zed_wrapper-2] 1654704475.687053773 [zed2.zed_node] [INFO]  * Initial pose: [0,0,0,0,0,0,]
[zed_wrapper-2] 1654704475.687137073 [zed2.zed_node] [INFO]  * Area Memory: FALSE
[zed_wrapper-2] 1654704475.687195507 [zed2.zed_node] [INFO]  * Area Memory DB: 
[zed_wrapper-2] 1654704475.687324089 [zed2.zed_node] [INFO]  * IMU Fusion [not for ZED]: TRUE
[zed_wrapper-2] 1654704475.687406588 [zed2.zed_node] [INFO]  * Floor Alignment: FALSE
[zed_wrapper-2] 1654704475.687459455 [zed2.zed_node] [INFO]  * Init Odometry with first valid pose data: TRUE
[zed_wrapper-2] 1654704475.687516929 [zed2.zed_node] [INFO]  * 2D mode: FALSE
[zed_wrapper-2] 1654704475.687567811 [zed2.zed_node] [INFO]  * Pose/Odometry QoS History: KEEP_LAST
[zed_wrapper-2] 1654704475.687608165 [zed2.zed_node] [INFO]  * Pose/Odometry QoS History depth: 1
[zed_wrapper-2] 1654704475.687647815 [zed2.zed_node] [INFO]  * Pose/Odometry QoS Reliability: RELIABLE
[zed_wrapper-2] 1654704475.687699785 [zed2.zed_node] [INFO]  * Pose/Odometry QoS Durability: TRANSIENT_LOCAL
[zed_wrapper-2] 1654704475.687727274 [zed2.zed_node] [INFO] *** SENSORS STACK parameters ***
[zed_wrapper-2] 1654704475.687777804 [zed2.zed_node] [INFO]  * Sensors Camera Sync: FALSE
[zed_wrapper-2] 1654704475.687842415 [zed2.zed_node] [INFO]  * Sensors publishing rate: 200 Hz
[zed_wrapper-2] 1654704475.687892049 [zed2.zed_node] [INFO]  * Sensors QoS History: KEEP_LAST
[zed_wrapper-2] 1654704475.687932083 [zed2.zed_node] [INFO]  * Sensors QoS History depth: 10
[zed_wrapper-2] 1654704475.687993333 [zed2.zed_node] [INFO]  * Sensors QoS Reliability: RELIABLE
[zed_wrapper-2] 1654704475.688033879 [zed2.zed_node] [INFO]  * Sensors QoS Durability: VOLATILE
[zed_wrapper-2] 1654704475.688059256 [zed2.zed_node] [INFO] *** Spatial Mapping parameters ***
[zed_wrapper-2] 1654704475.688101530 [zed2.zed_node] [INFO]  * Spatial Mapping Enabled: FALSE
[zed_wrapper-2] 1654704475.688182013 [zed2.zed_node] [INFO]  * Spatial Mapping resolution [m]: 0.1
[zed_wrapper-2] 1654704475.688256321 [zed2.zed_node] [INFO]  * 3D Max Mapping range [m]: 20
[zed_wrapper-2] 1654704475.688313027 [zed2.zed_node] [INFO]  * Map publishing rate [Hz]: 0.1
[zed_wrapper-2] 1654704475.688356229 [zed2.zed_node] [INFO]  * Sensors QoS History: KEEP_LAST
[zed_wrapper-2] 1654704475.688394919 [zed2.zed_node] [INFO]  * Sensors QoS History depth: 1
[zed_wrapper-2] 1654704475.688437064 [zed2.zed_node] [INFO]  * Sensors QoS Reliability: RELIABLE
[zed_wrapper-2] 1654704475.688486250 [zed2.zed_node] [INFO]  * Sensors QoS Durability: TRANSIENT_LOCAL
[zed_wrapper-2] 1654704475.688512492 [zed2.zed_node] [INFO] *** OBJECT DETECTION parameters ***
[zed_wrapper-2] 1654704475.688561934 [zed2.zed_node] [INFO]  * Object Detection enabled: FALSE
[zed_wrapper-2] 1654704475.688643697 [zed2.zed_node] [INFO]  * OD min. confidence: 50
[zed_wrapper-2] 1654704475.688725653 [zed2.zed_node] [INFO]  * OD tracking: TRUE
[zed_wrapper-2] 1654704475.688822169 [zed2.zed_node] [INFO]  * Object Detection model: 0 - MULTI CLASS BOX FAST
[zed_wrapper-2] 1654704475.688919197 [zed2.zed_node] [INFO]  * Object Filtering mode: 1 - NMS 3D
[zed_wrapper-2] 1654704475.689009761 [zed2.zed_node] [INFO]  * MultiClassBox people: TRUE
[zed_wrapper-2] 1654704475.689082596 [zed2.zed_node] [INFO]  * MultiClassBox vehicles: TRUE
[zed_wrapper-2] 1654704475.689169320 [zed2.zed_node] [INFO]  * MultiClassBox bags: TRUE
[zed_wrapper-2] 1654704475.689226090 [zed2.zed_node] [INFO]  * MultiClassBox animals: TRUE
[zed_wrapper-2] 1654704475.689279820 [zed2.zed_node] [INFO]  * MultiClassBox electronics: TRUE
[zed_wrapper-2] 1654704475.689360976 [zed2.zed_node] [INFO]  * MultiClassBox fruits and vegetables: TRUE
[zed_wrapper-2] 1654704475.689431123 [zed2.zed_node] [INFO]  * MultiClassBox sport-related objects: TRUE
[zed_wrapper-2] 1654704475.689519895 [zed2.zed_node] [INFO]  * Body format: 1 - POSE_34
[zed_wrapper-2] 1654704475.689548344 [zed2.zed_node] [INFO]  * Skeleton fitting: TRUE (forced by `object_detection.body_format`)
[zed_wrapper-2] 1654704475.689599450 [zed2.zed_node] [INFO]  * Obj. Det. QoS History: KEEP_LAST
[zed_wrapper-2] 1654704475.689650972 [zed2.zed_node] [INFO]  * Obj. Det. QoS History depth: 1
[zed_wrapper-2] 1654704475.689702302 [zed2.zed_node] [INFO]  * Obj. Det. QoS Reliability: BEST_EFFORT
[zed_wrapper-2] 1654704475.689807235 [zed2.zed_node] [INFO]  * Obj. Det. QoS Durability: VOLATILE
[zed_wrapper-2] 1654704475.690048525 [zed2.zed_node] [INFO] *** SERVICES ***
[zed_wrapper-2] 1654704475.691438408 [zed2.zed_node] [INFO]  * '/zed2/zed_node/reset_odometry'
[zed_wrapper-2] 1654704475.692501750 [zed2.zed_node] [INFO]  * '/zed2/zed_node/reset_pos_tracking'
[zed_wrapper-2] 1654704475.694138812 [zed2.zed_node] [INFO]  * '/zed2/zed_node/set_pose'
[zed_wrapper-2] 1654704475.695380945 [zed2.zed_node] [INFO]  * '/zed2/zed_node/enable_obj_det'
[zed_wrapper-2] 1654704475.696264695 [zed2.zed_node] [INFO]  * '/zed2/zed_node/enable_mapping'
[zed_wrapper-2] 1654704475.696990198 [zed2.zed_node] [INFO]  * '/zed2/zed_node/start_svo_rec'
[zed_wrapper-2] 1654704475.697824025 [zed2.zed_node] [INFO]  * '/zed2/zed_node/stop_svo_rec'
[zed_wrapper-2] 1654704475.698523575 [zed2.zed_node] [INFO]  * '/zed2/zed_node/toggle_svo_pause'
[zed_wrapper-2] 1654704475.698606779 [zed2.zed_node] [INFO] ***** STARTING CAMERA *****
[zed_wrapper-2] 1654704475.698636924 [zed2.zed_node] [INFO] SDK Version: 3.7.2 - Build 43767_1578be58
[zed_wrapper-2] 1654704475.703901597 [zed2.zed_node] [INFO] *** CAMERA OPENING ***
[zed_wrapper-2] [ZED][INFO] Logging level INFO
[zed_wrapper-2] [ZED][INFO] [Init]  Depth mode: NONE
[zed_wrapper-2] [ZED][WARNING] libhid error: -4
[zed_wrapper-2] [ZED][WARNING] [Sensors] MCU module not responding. Attempt module reboot
[zed_wrapper-2] [ZED][MCU] Permissions denied : can't open device. Make sure you have installed udev rules or use sudo
[zed_wrapper-2] [ZED][WARNING] [ZED] Sensors could not be initialized. Some modules (AI-based) might not work correctly.
[zed_wrapper-2] [ZED][INFO] [Init]  Camera successfully opened.
[zed_wrapper-2] [ZED][INFO] [Init]  Camera FW version: 1523
[zed_wrapper-2] [ZED][INFO] [Init]  Video mode: VGA@60
[zed_wrapper-2] 1654704476.999105480 [zed2.zed_node] [INFO] ZED SDK running on GPU #0
[zed_wrapper-2] 1654704476.999275215 [zed2.zed_node] [INFO]  * Camera Model	-> ZED 2
[zed_wrapper-2] 1654704476.999319761 [zed2.zed_node] [INFO]  * Serial Number	-> 28813151
[zed_wrapper-2] 1654704476.999394388 [zed2.zed_node] [INFO]  * Input type	-> USB input type
[zed_wrapper-2] 1654704476.999435190 [zed2.zed_node] [INFO]  * Camera FW Version -> 1523
[zed_wrapper-2] 1654704476.999489400 [zed2.zed_node] [INFO]  * Sensors FW Version -> 0
[zed_wrapper-2] 1654704476.999676320 [zed2.zed_node] [INFO] *** TF FRAMES ***
[zed_wrapper-2] 1654704476.999713826 [zed2.zed_node] [INFO]  * Map			-> map
[zed_wrapper-2] 1654704476.999740931 [zed2.zed_node] [INFO]  * Odometry			-> vis_odom
[zed_wrapper-2] 1654704476.999766532 [zed2.zed_node] [INFO]  * Base			-> base_link
[zed_wrapper-2] 1654704476.999821478 [zed2.zed_node] [INFO]  * Camera			-> zed2_camera_center
[zed_wrapper-2] 1654704476.999848392 [zed2.zed_node] [INFO]  * Left			-> zed2_left_camera_frame
[zed_wrapper-2] 1654704476.999905834 [zed2.zed_node] [INFO]  * Left Optical		-> zed2_left_camera_optical_frame
[zed_wrapper-2] 1654704476.999952684 [zed2.zed_node] [INFO]  * RGB			-> zed2_left_camera_frame
[zed_wrapper-2] 1654704477.000003278 [zed2.zed_node] [INFO]  * RGB Optical		-> zed2_left_camera_frame
[zed_wrapper-2] 1654704477.000049616 [zed2.zed_node] [INFO]  * Right			-> zed2_right_camera_frame
[zed_wrapper-2] 1654704477.000077009 [zed2.zed_node] [INFO]  * Right Optical		-> zed2_right_camera_optical_frame
[zed_wrapper-2] 1654704477.000103730 [zed2.zed_node] [INFO]  * Depth			-> zed2_left_camera_frame
[zed_wrapper-2] 1654704477.000128467 [zed2.zed_node] [INFO]  * Depth Optical		-> zed2_left_camera_optical_frame
[zed_wrapper-2] 1654704477.000153013 [zed2.zed_node] [INFO]  * Point Cloud		-> zed2_left_camera_optical_frame
[zed_wrapper-2] 1654704477.000178198 [zed2.zed_node] [INFO]  * Disparity		-> zed2_left_camera_frame
[zed_wrapper-2] 1654704477.000202551 [zed2.zed_node] [INFO]  * Disparity Optical	-> zed2_left_camera_optical_frame
[zed_wrapper-2] 1654704477.000226552 [zed2.zed_node] [INFO]  * Confidence		-> zed2_left_camera_frame
[zed_wrapper-2] 1654704477.000251609 [zed2.zed_node] [INFO]  * Confidence Optical	-> zed2_left_camera_optical_frame
[zed_wrapper-2] 1654704477.000302075 [zed2.zed_node] [INFO]  * IMU			-> zed2_imu_link
[zed_wrapper-2] 1654704477.000356221 [zed2.zed_node] [INFO]  * Barometer		-> zed2_camera_center
[zed_wrapper-2] 1654704477.000404607 [zed2.zed_node] [INFO]  * Magnetometer		-> zed2_imu_link
[zed_wrapper-2] 1654704477.000430624 [zed2.zed_node] [INFO]  * Left Temperature	-> zed2_left_camera_frame
[zed_wrapper-2] 1654704477.000456737 [zed2.zed_node] [INFO]  * Right Temperature	-> zed2_right_camera_frame
[zed_wrapper-2] 1654704477.000490147 [zed2.zed_node] [INFO] *** PUBLISHED TOPICS ***
[zed_wrapper-2] 1654704477.008524442 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb/image_rect_color
[zed_wrapper-2] 1654704477.009803633 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb/image_rect_gray
[zed_wrapper-2] 1654704477.009889684 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb/camera_info
[zed_wrapper-2] 1654704477.012620169 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb_raw/image_raw_color
[zed_wrapper-2] 1654704477.013817596 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb_raw/image_raw_gray
[zed_wrapper-2] 1654704477.013912608 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb_raw/camera_info
[zed_wrapper-2] 1654704477.015314812 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left/image_rect_color
[zed_wrapper-2] 1654704477.016625204 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left/image_rect_gray
[zed_wrapper-2] 1654704477.016719320 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left/camera_info
[zed_wrapper-2] 1654704477.019271813 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left_raw/image_raw_color
[zed_wrapper-2] 1654704477.020378452 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left_raw/image_raw_gray
[zed_wrapper-2] 1654704477.020453207 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left_raw/camera_info
[zed_wrapper-2] 1654704477.021672779 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right/image_rect_color
[zed_wrapper-2] 1654704477.022970179 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right/image_rect_gray
[zed_wrapper-2] 1654704477.023096776 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right/camera_info
[zed_wrapper-2] 1654704477.024331037 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right_raw/image_raw_color
[zed_wrapper-2] 1654704477.025462029 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right_raw/image_raw_gray
[zed_wrapper-2] 1654704477.025529200 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right_raw/camera_info
[zed_wrapper-2] 1654704477.026111113 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/stereo/image_rect_color
[zed_wrapper-2] 1654704477.026614878 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/stereo_raw/image_raw_color
[zed_wrapper-2] 1654704477.027281275 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/imu/data
[zed_wrapper-2] 1654704477.027765840 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/imu/data_raw
[zed_wrapper-2] 1654704477.028418956 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/temperature/imu
[zed_wrapper-2] 1654704477.029173132 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/imu/mag
[zed_wrapper-2] 1654704477.029818951 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/atm_press
[zed_wrapper-2] 1654704477.030282939 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/temperature/left
[zed_wrapper-2] 1654704477.030759951 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/temperature/right
[zed_wrapper-2] 1654704477.031760666 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left_cam_imu_transform [LATCHED]
[zed_wrapper-2] 1654704477.031849950 [zed2.zed_node] [INFO] Camera-IMU Translation: 
[zed_wrapper-2]  0 0 0
[zed_wrapper-2] 1654704477.031928481 [zed2.zed_node] [INFO] Camera-IMU Rotation: 
[zed_wrapper-2]  7FD95DDB78
[zed_wrapper-2] 1.000000 0.000000 0.000000
[zed_wrapper-2] 0.000000 1.000000 0.000000
[zed_wrapper-2] 0.000000 0.000000 1.000000
[zed_wrapper-2] 
[zed_wrapper-2] 1654704477.032022597 [zed2.zed_node] [INFO] Published static TF: 'zed2_imu_link' -> 'zed2_left_camera_frame'
[ERROR] [zed_wrapper-2]: process has died [pid 5400, exit code -11, cmd '/code/ros2_ws/install/zed_wrapper/lib/zed_wrapper/zed_wrapper --ros-args -r __node:=zed_node -r __ns:=/zed2 --params-file /code/ros2_ws/install/zed_wrapper/share/zed_wrapper/config/common.yaml --params-file /code/ros2_ws/install/zed_wrapper/share/zed_wrapper/config/zed2.yaml --params-file /tmp/launch_params_shgbxu9a'].

### As for the position tracking problem:

ros2 launch zed_wrapper zed2.launch.py 
[INFO] [launch]: All log files can be found below /root/.ros/log/2022-06-08-16-31-09-403505-ubuntu-8280
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [robot_state_publisher-1]: process started with pid [8284]
[INFO] [zed_wrapper-2]: process started with pid [8285]
[robot_state_publisher-1] Parsing robot urdf xml string.
[robot_state_publisher-1] Link zed2_base_link had 1 children
[robot_state_publisher-1] Link zed2_camera_center had 4 children
[robot_state_publisher-1] Link zed2_baro_link had 0 children
[robot_state_publisher-1] Link zed2_left_camera_frame had 2 children
[robot_state_publisher-1] Link zed2_left_camera_optical_frame had 0 children
[robot_state_publisher-1] Link zed2_temp_left_link had 0 children
[robot_state_publisher-1] Link zed2_mag_link had 0 children
[robot_state_publisher-1] Link zed2_right_camera_frame had 2 children
[robot_state_publisher-1] Link zed2_right_camera_optical_frame had 0 children
[robot_state_publisher-1] Link zed2_temp_right_link had 0 children
[robot_state_publisher-1] 1654705869.905225481 [zed2.zed_state_publisher] [INFO] got segment base_link
[robot_state_publisher-1] 1654705869.905502101 [zed2.zed_state_publisher] [INFO] got segment zed2_baro_link
[robot_state_publisher-1] 1654705869.905539638 [zed2.zed_state_publisher] [INFO] got segment zed2_base_link
[robot_state_publisher-1] 1654705869.905566167 [zed2.zed_state_publisher] [INFO] got segment zed2_camera_center
[robot_state_publisher-1] 1654705869.905588632 [zed2.zed_state_publisher] [INFO] got segment zed2_left_camera_frame
[robot_state_publisher-1] 1654705869.905612185 [zed2.zed_state_publisher] [INFO] got segment zed2_left_camera_optical_frame
[robot_state_publisher-1] 1654705869.905634586 [zed2.zed_state_publisher] [INFO] got segment zed2_mag_link
[robot_state_publisher-1] 1654705869.905670044 [zed2.zed_state_publisher] [INFO] got segment zed2_right_camera_frame
[robot_state_publisher-1] 1654705869.905693213 [zed2.zed_state_publisher] [INFO] got segment zed2_right_camera_optical_frame
[robot_state_publisher-1] 1654705869.905715102 [zed2.zed_state_publisher] [INFO] got segment zed2_temp_left_link
[robot_state_publisher-1] 1654705869.905737119 [zed2.zed_state_publisher] [INFO] got segment zed2_temp_right_link
[zed_wrapper-2] 1654705869.980049918 [zed2.zed_node] [INFO] ********************************
[zed_wrapper-2] 1654705869.980391661 [zed2.zed_node] [INFO]       ZED Camera Component 
[zed_wrapper-2] 1654705869.980453904 [zed2.zed_node] [INFO] ********************************
[zed_wrapper-2] 1654705869.980486257 [zed2.zed_node] [INFO]  * namespace: /zed2
[zed_wrapper-2] 1654705869.980533203 [zed2.zed_node] [INFO]  * node name: zed_node
[zed_wrapper-2] 1654705869.980555540 [zed2.zed_node] [INFO] ********************************
[zed_wrapper-2] 1654705869.980596982 [zed2.zed_node] [INFO] *** DEBUG parameters ***
[zed_wrapper-2] 1654705869.980675641 [zed2.zed_node] [INFO]  * Debug mode: FALSE
[zed_wrapper-2] 1654705869.980718875 [zed2.zed_node] [INFO] *** GENERAL parameters ***
[zed_wrapper-2] 1654705869.980805439 [zed2.zed_node] [INFO]  * Camera model: zed2 - ZED 2
[zed_wrapper-2] 1654705869.980875202 [zed2.zed_node] [INFO]  * SDK Verbose: 1
[zed_wrapper-2] 1654705869.980938373 [zed2.zed_node] [INFO]  * SVO: live
[zed_wrapper-2] 1654705869.981034889 [zed2.zed_node] [INFO]  * SVO Loop: FALSE
[zed_wrapper-2] 1654705869.981127757 [zed2.zed_node] [INFO]  * SVO Realtime: FALSE
[zed_wrapper-2] 1654705869.981208145 [zed2.zed_node] [INFO]  * Camera name: zed2
[zed_wrapper-2] 1654705869.981271539 [zed2.zed_node] [INFO]  * Camera ID: 0
[zed_wrapper-2] 1654705869.981336438 [zed2.zed_node] [INFO]  * Camera SN: 0
[zed_wrapper-2] 1654705869.981396569 [zed2.zed_node] [INFO]  * Camera timeout [sec]: 5
[zed_wrapper-2] 1654705869.981455323 [zed2.zed_node] [INFO]  * Camera reconnection temptatives: 5
[zed_wrapper-2] 1654705869.981563584 [zed2.zed_node] [INFO]  * Camera framerate: 60
[zed_wrapper-2] 1654705869.981640355 [zed2.zed_node] [INFO]  * GPU ID: -1
[zed_wrapper-2] 1654705869.981735048 [zed2.zed_node] [INFO]  * Camera resolution: 3 - VGA
[zed_wrapper-2] 1654705869.981790410 [zed2.zed_node] [INFO]  * Camera self calibration: TRUE
[zed_wrapper-2] 1654705869.981840908 [zed2.zed_node] [INFO]  * Camera flip: FALSE
[zed_wrapper-2] 1654705869.981895823 [zed2.zed_node] [INFO]  * [DYN] Publish framerate [Hz]: 40 
[zed_wrapper-2] 1654705869.981931760 [zed2.zed_node] [INFO] *** VIDEO parameters ***
[zed_wrapper-2] 1654705869.981993235 [zed2.zed_node] [INFO]  * Use old extrinsic parameters: 0
[zed_wrapper-2] 1654705869.982057174 [zed2.zed_node] [INFO]  * [DYN] Image downsample factor: 1 
[zed_wrapper-2] 1654705869.982148570 [zed2.zed_node] [INFO]  * [DYN] Brightness: 4
[zed_wrapper-2] 1654705869.982249950 [zed2.zed_node] [INFO]  * [DYN] Contrast: 4
[zed_wrapper-2] 1654705869.982311105 [zed2.zed_node] [INFO]  * [DYN] Hue: 0
[zed_wrapper-2] 1654705869.982382948 [zed2.zed_node] [INFO]  * [DYN] Saturation: 4
[zed_wrapper-2] 1654705869.982489768 [zed2.zed_node] [INFO]  * [DYN] Sharpness: 4
[zed_wrapper-2] 1654705869.982554123 [zed2.zed_node] [INFO]  * [DYN] Gamma: 8
[zed_wrapper-2] 1654705869.982611758 [zed2.zed_node] [INFO]  * [DYN] Auto Exposure/Gain: TRUE
[zed_wrapper-2] 1654705869.982665424 [zed2.zed_node] [INFO]  * [DYN] Exposure: 80
[zed_wrapper-2] 1654705869.982761236 [zed2.zed_node] [INFO]  * [DYN] Gain: 80
[zed_wrapper-2] 1654705869.982836951 [zed2.zed_node] [INFO]  * [DYN] Auto White Balance: TRUE
[zed_wrapper-2] 1654705869.982920955 [zed2.zed_node] [INFO]  * [DYN] White Balance Temperature: 42
[zed_wrapper-2] 1654705869.982977502 [zed2.zed_node] [INFO]  * Video QoS History: KEEP_LAST
[zed_wrapper-2] 1654705869.983029984 [zed2.zed_node] [INFO]  * Video QoS History depth: 1
[zed_wrapper-2] 1654705869.983096867 [zed2.zed_node] [INFO]  * Video QoS Reliability: RELIABLE
[zed_wrapper-2] 1654705869.983148869 [zed2.zed_node] [INFO]  * Video QoS Durability: VOLATILE
[zed_wrapper-2] 1654705869.983174950 [zed2.zed_node] [INFO] *** DEPTH parameters ***
[zed_wrapper-2] 1654705869.983227848 [zed2.zed_node] [INFO]  * Depth quality: 1 - PERFORMANCE
[zed_wrapper-2] 1654705869.983302124 [zed2.zed_node] [INFO]  * Depth downsample factor: 0.1 
[zed_wrapper-2] 1654705869.983390128 [zed2.zed_node] [INFO]  * Min depth [m]: 0.7
[zed_wrapper-2] 1654705869.983452274 [zed2.zed_node] [INFO]  * Max depth [m]: 20
[zed_wrapper-2] 1654705869.983513653 [zed2.zed_node] [INFO]  * Depth Sensing Mode: 0 - STANDARD
[zed_wrapper-2] 1654705869.983593176 [zed2.zed_node] [INFO]  * Depth Stabilization: FALSE
[zed_wrapper-2] 1654705869.983648059 [zed2.zed_node] [INFO]  * OpenNI mode (16bit point cloud): FALSE
[zed_wrapper-2] 1654705869.983712350 [zed2.zed_node] [INFO]  * [DYN] Point cloud rate [Hz]: 0.1
[zed_wrapper-2] 1654705869.983770560 [zed2.zed_node] [INFO]  * [DYN] Depth Confidence: 50
[zed_wrapper-2] 1654705869.983828163 [zed2.zed_node] [INFO]  * [DYN] Depth Texture Confidence: 100
[zed_wrapper-2] 1654705869.983882021 [zed2.zed_node] [INFO]  * [DYN] Remove saturated areas: TRUE
[zed_wrapper-2] 1654705869.983933191 [zed2.zed_node] [INFO]  * Depth QoS History: KEEP_LAST
[zed_wrapper-2] 1654705869.983998410 [zed2.zed_node] [INFO]  * Depth QoS History depth: 1
[zed_wrapper-2] 1654705869.984050284 [zed2.zed_node] [INFO]  * Depth QoS Reliability: RELIABLE
[zed_wrapper-2] 1654705869.984172306 [zed2.zed_node] [INFO]  * Depth QoS Durability: VOLATILE
[zed_wrapper-2] 1654705869.984200467 [zed2.zed_node] [INFO] *** POSITIONAL TRACKING parameters ***
[zed_wrapper-2] 1654705869.984274262 [zed2.zed_node] [INFO]  * Positional tracking enabled: FALSE
[zed_wrapper-2] 1654705869.984350841 [zed2.zed_node] [INFO]  * Base frame id: base_link
[zed_wrapper-2] 1654705869.984412636 [zed2.zed_node] [INFO]  * Map frame id: map
[zed_wrapper-2] 1654705869.984469758 [zed2.zed_node] [INFO]  * Odometry frame id: vis_odom
[zed_wrapper-2] 1654705869.984534913 [zed2.zed_node] [INFO]  * Broadcast Odometry TF: TRUE
[zed_wrapper-2] 1654705869.984593956 [zed2.zed_node] [INFO]  * Broadcast Pose TF: TRUE
[zed_wrapper-2] 1654705869.984658951 [zed2.zed_node] [INFO]  * Broadcast Static IMU TF [not for ZED]: TRUE
[zed_wrapper-2] 1654705869.984721897 [zed2.zed_node] [INFO]  * [DYN] Path publishing rate: 10
[zed_wrapper-2] 1654705869.984781196 [zed2.zed_node] [INFO]  * Path history lenght: 10
[zed_wrapper-2] 1654705869.984844719 [zed2.zed_node] [INFO]  * Initial pose: [0,0,0,0,0,0,]
[zed_wrapper-2] 1654705869.984921778 [zed2.zed_node] [INFO]  * Area Memory: FALSE
[zed_wrapper-2] 1654705869.984990261 [zed2.zed_node] [INFO]  * Area Memory DB: 
[zed_wrapper-2] 1654705869.985049368 [zed2.zed_node] [INFO]  * IMU Fusion [not for ZED]: TRUE
[zed_wrapper-2] 1654705869.985106522 [zed2.zed_node] [INFO]  * Floor Alignment: FALSE
[zed_wrapper-2] 1654705869.985178973 [zed2.zed_node] [INFO]  * Init Odometry with first valid pose data: TRUE
[zed_wrapper-2] 1654705869.985295906 [zed2.zed_node] [INFO]  * 2D mode: FALSE
[zed_wrapper-2] 1654705869.985350309 [zed2.zed_node] [INFO]  * Pose/Odometry QoS History: KEEP_LAST
[zed_wrapper-2] 1654705869.985398919 [zed2.zed_node] [INFO]  * Pose/Odometry QoS History depth: 1
[zed_wrapper-2] 1654705869.985448777 [zed2.zed_node] [INFO]  * Pose/Odometry QoS Reliability: RELIABLE
[zed_wrapper-2] 1654705869.985504139 [zed2.zed_node] [INFO]  * Pose/Odometry QoS Durability: TRANSIENT_LOCAL
[zed_wrapper-2] 1654705869.985534221 [zed2.zed_node] [INFO] *** SENSORS STACK parameters ***
[zed_wrapper-2] 1654705869.985607408 [zed2.zed_node] [INFO]  * Sensors Camera Sync: FALSE
[zed_wrapper-2] 1654705869.985663762 [zed2.zed_node] [INFO]  * Sensors publishing rate: 200 Hz
[zed_wrapper-2] 1654705869.985761303 [zed2.zed_node] [INFO]  * Sensors QoS History: KEEP_LAST
[zed_wrapper-2] 1654705869.985825273 [zed2.zed_node] [INFO]  * Sensors QoS History depth: 10
[zed_wrapper-2] 1654705869.985872987 [zed2.zed_node] [INFO]  * Sensors QoS Reliability: RELIABLE
[zed_wrapper-2] 1654705869.985957695 [zed2.zed_node] [INFO]  * Sensors QoS Durability: VOLATILE
[zed_wrapper-2] 1654705869.985986272 [zed2.zed_node] [INFO] *** Spatial Mapping parameters ***
[zed_wrapper-2] 1654705869.986032354 [zed2.zed_node] [INFO]  * Spatial Mapping Enabled: FALSE
[zed_wrapper-2] 1654705869.986124358 [zed2.zed_node] [INFO]  * Spatial Mapping resolution [m]: 0.1
[zed_wrapper-2] 1654705869.986213514 [zed2.zed_node] [INFO]  * 3D Max Mapping range [m]: 20
[zed_wrapper-2] 1654705869.986265453 [zed2.zed_node] [INFO]  * Map publishing rate [Hz]: 0.1
[zed_wrapper-2] 1654705869.986320111 [zed2.zed_node] [INFO]  * Sensors QoS History: KEEP_LAST
[zed_wrapper-2] 1654705869.986370033 [zed2.zed_node] [INFO]  * Sensors QoS History depth: 1
[zed_wrapper-2] 1654705869.986489654 [zed2.zed_node] [INFO]  * Sensors QoS Reliability: RELIABLE
[zed_wrapper-2] 1654705869.986544377 [zed2.zed_node] [INFO]  * Sensors QoS Durability: TRANSIENT_LOCAL
[zed_wrapper-2] 1654705869.986572506 [zed2.zed_node] [INFO] *** OBJECT DETECTION parameters ***
[zed_wrapper-2] 1654705869.986625244 [zed2.zed_node] [INFO]  * Object Detection enabled: FALSE
[zed_wrapper-2] 1654705869.986693183 [zed2.zed_node] [INFO]  * OD min. confidence: 50
[zed_wrapper-2] 1654705869.986780387 [zed2.zed_node] [INFO]  * OD tracking: TRUE
[zed_wrapper-2] 1654705869.986866855 [zed2.zed_node] [INFO]  * Object Detection model: 0 - MULTI CLASS BOX FAST
[zed_wrapper-2] 1654705869.986953354 [zed2.zed_node] [INFO]  * Object Filtering mode: 1 - NMS 3D
[zed_wrapper-2] 1654705869.987043278 [zed2.zed_node] [INFO]  * MultiClassBox people: TRUE
[zed_wrapper-2] 1654705869.987099985 [zed2.zed_node] [INFO]  * MultiClassBox vehicles: TRUE
[zed_wrapper-2] 1654705869.987163604 [zed2.zed_node] [INFO]  * MultiClassBox bags: TRUE
[zed_wrapper-2] 1654705869.987219222 [zed2.zed_node] [INFO]  * MultiClassBox animals: TRUE
[zed_wrapper-2] 1654705869.987276153 [zed2.zed_node] [INFO]  * MultiClassBox electronics: TRUE
[zed_wrapper-2] 1654705869.987331611 [zed2.zed_node] [INFO]  * MultiClassBox fruits and vegetables: TRUE
[zed_wrapper-2] 1654705869.987399262 [zed2.zed_node] [INFO]  * MultiClassBox sport-related objects: TRUE
[zed_wrapper-2] 1654705869.987466017 [zed2.zed_node] [INFO]  * Body format: 1 - POSE_34
[zed_wrapper-2] 1654705869.987542372 [zed2.zed_node] [INFO]  * Skeleton fitting: TRUE (forced by `object_detection.body_format`)
[zed_wrapper-2] 1654705869.987597478 [zed2.zed_node] [INFO]  * Obj. Det. QoS History: KEEP_LAST
[zed_wrapper-2] 1654705869.987653641 [zed2.zed_node] [INFO]  * Obj. Det. QoS History depth: 1
[zed_wrapper-2] 1654705869.987730412 [zed2.zed_node] [INFO]  * Obj. Det. QoS Reliability: BEST_EFFORT
[zed_wrapper-2] 1654705869.987782735 [zed2.zed_node] [INFO]  * Obj. Det. QoS Durability: VOLATILE
[zed_wrapper-2] 1654705869.987976023 [zed2.zed_node] [INFO] *** SERVICES ***
[zed_wrapper-2] 1654705869.989368051 [zed2.zed_node] [INFO]  * '/zed2/zed_node/reset_odometry'
[zed_wrapper-2] 1654705869.989972014 [zed2.zed_node] [INFO]  * '/zed2/zed_node/reset_pos_tracking'
[zed_wrapper-2] 1654705869.991067293 [zed2.zed_node] [INFO]  * '/zed2/zed_node/set_pose'
[zed_wrapper-2] 1654705869.991764156 [zed2.zed_node] [INFO]  * '/zed2/zed_node/enable_obj_det'
[zed_wrapper-2] 1654705869.992255153 [zed2.zed_node] [INFO]  * '/zed2/zed_node/enable_mapping'
[zed_wrapper-2] 1654705869.992872844 [zed2.zed_node] [INFO]  * '/zed2/zed_node/start_svo_rec'
[zed_wrapper-2] 1654705869.993334560 [zed2.zed_node] [INFO]  * '/zed2/zed_node/stop_svo_rec'
[zed_wrapper-2] 1654705869.993814261 [zed2.zed_node] [INFO]  * '/zed2/zed_node/toggle_svo_pause'
[zed_wrapper-2] 1654705869.993857207 [zed2.zed_node] [INFO] ***** STARTING CAMERA *****
[zed_wrapper-2] 1654705869.993887160 [zed2.zed_node] [INFO] SDK Version: 3.7.2 - Build 43767_1578be58
[zed_wrapper-2] 1654705869.998612165 [zed2.zed_node] [INFO] *** CAMERA OPENING ***
[zed_wrapper-2] [ZED][INFO] Logging level INFO
[zed_wrapper-2] [ZED][INFO] [Init]  Depth mode: PERFORMANCE
[zed_wrapper-2] [ZED][WARNING] libhid error: -4
[zed_wrapper-2] [ZED][WARNING] [Sensors] MCU module not responding. Attempt module reboot
[zed_wrapper-2] [ZED][MCU] Permissions denied : can't open device. Make sure you have installed udev rules or use sudo
[zed_wrapper-2] [ZED][WARNING] [ZED] Sensors could not be initialized. Some modules (AI-based) might not work correctly.
[zed_wrapper-2] [ZED][INFO] [Init]  Camera successfully opened.
[zed_wrapper-2] [ZED][INFO] [Init]  Camera FW version: 1523
[zed_wrapper-2] [ZED][INFO] [Init]  Video mode: VGA@60
[zed_wrapper-2] 1654705871.301208166 [zed2.zed_node] [INFO] ZED SDK running on GPU #0
[zed_wrapper-2] 1654705871.301351564 [zed2.zed_node] [INFO]  * Camera Model	-> ZED 2
[zed_wrapper-2] 1654705871.301384109 [zed2.zed_node] [INFO]  * Serial Number	-> 28813151
[zed_wrapper-2] 1654705871.301449840 [zed2.zed_node] [INFO]  * Input type	-> USB input type
[zed_wrapper-2] 1654705871.301488530 [zed2.zed_node] [INFO]  * Camera FW Version -> 1523
[zed_wrapper-2] 1654705871.301541460 [zed2.zed_node] [INFO]  * Sensors FW Version -> 0
[zed_wrapper-2] 1654705871.301719452 [zed2.zed_node] [INFO] *** TF FRAMES ***
[zed_wrapper-2] 1654705871.301754557 [zed2.zed_node] [INFO]  * Map			-> map
[zed_wrapper-2] 1654705871.301779710 [zed2.zed_node] [INFO]  * Odometry			-> vis_odom
[zed_wrapper-2] 1654705871.301804224 [zed2.zed_node] [INFO]  * Base			-> base_link
[zed_wrapper-2] 1654705871.301828129 [zed2.zed_node] [INFO]  * Camera			-> zed2_camera_center
[zed_wrapper-2] 1654705871.301852450 [zed2.zed_node] [INFO]  * Left			-> zed2_left_camera_frame
[zed_wrapper-2] 1654705871.301878755 [zed2.zed_node] [INFO]  * Left Optical		-> zed2_left_camera_optical_frame
[zed_wrapper-2] 1654705871.301929669 [zed2.zed_node] [INFO]  * RGB			-> zed2_left_camera_frame
[zed_wrapper-2] 1654705871.301976871 [zed2.zed_node] [INFO]  * RGB Optical		-> zed2_left_camera_frame
[zed_wrapper-2] 1654705871.302032041 [zed2.zed_node] [INFO]  * Right			-> zed2_right_camera_frame
[zed_wrapper-2] 1654705871.302058891 [zed2.zed_node] [INFO]  * Right Optical		-> zed2_right_camera_optical_frame
[zed_wrapper-2] 1654705871.302085580 [zed2.zed_node] [INFO]  * Depth			-> zed2_left_camera_frame
[zed_wrapper-2] 1654705871.302111629 [zed2.zed_node] [INFO]  * Depth Optical		-> zed2_left_camera_optical_frame
[zed_wrapper-2] 1654705871.302148623 [zed2.zed_node] [INFO]  * Point Cloud		-> zed2_left_camera_optical_frame
[zed_wrapper-2] 1654705871.302174000 [zed2.zed_node] [INFO]  * Disparity		-> zed2_left_camera_frame
[zed_wrapper-2] 1654705871.302199761 [zed2.zed_node] [INFO]  * Disparity Optical	-> zed2_left_camera_optical_frame
[zed_wrapper-2] 1654705871.302224530 [zed2.zed_node] [INFO]  * Confidence		-> zed2_left_camera_frame
[zed_wrapper-2] 1654705871.302249971 [zed2.zed_node] [INFO]  * Confidence Optical	-> zed2_left_camera_optical_frame
[zed_wrapper-2] 1654705871.302300597 [zed2.zed_node] [INFO]  * IMU			-> zed2_imu_link
[zed_wrapper-2] 1654705871.302355192 [zed2.zed_node] [INFO]  * Barometer		-> zed2_camera_center
[zed_wrapper-2] 1654705871.302405210 [zed2.zed_node] [INFO]  * Magnetometer		-> zed2_imu_link
[zed_wrapper-2] 1654705871.302481853 [zed2.zed_node] [INFO]  * Left Temperature	-> zed2_left_camera_frame
[zed_wrapper-2] 1654705871.302508926 [zed2.zed_node] [INFO]  * Right Temperature	-> zed2_right_camera_frame
[zed_wrapper-2] 1654705871.302548000 [zed2.zed_node] [INFO] *** PUBLISHED TOPICS ***
[zed_wrapper-2] 1654705871.310350067 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb/image_rect_color
[zed_wrapper-2] 1654705871.311942424 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb/image_rect_gray
[zed_wrapper-2] 1654705871.312035356 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb/camera_info
[zed_wrapper-2] 1654705871.313178958 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb_raw/image_raw_color
[zed_wrapper-2] 1654705871.314628685 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb_raw/image_raw_gray
[zed_wrapper-2] 1654705871.314709937 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/rgb_raw/camera_info
[zed_wrapper-2] 1654705871.316035978 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left/image_rect_color
[zed_wrapper-2] 1654705871.317235007 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left/image_rect_gray
[zed_wrapper-2] 1654705871.317306242 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left/camera_info
[zed_wrapper-2] 1654705871.318370192 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left_raw/image_raw_color
[zed_wrapper-2] 1654705871.319713642 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left_raw/image_raw_gray
[zed_wrapper-2] 1654705871.319792398 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left_raw/camera_info
[zed_wrapper-2] 1654705871.320759672 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right/image_rect_color
[zed_wrapper-2] 1654705871.321668223 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right/image_rect_gray
[zed_wrapper-2] 1654705871.321739682 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right/camera_info
[zed_wrapper-2] 1654705871.322891988 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right_raw/image_raw_color
[zed_wrapper-2] 1654705871.324126442 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right_raw/image_raw_gray
[zed_wrapper-2] 1654705871.324186957 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/right_raw/camera_info
[zed_wrapper-2] 1654705871.325267612 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/depth/depth_registered
[zed_wrapper-2] 1654705871.325335231 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/depth/camera_info
[zed_wrapper-2] 1654705871.325959226 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/depth/depth_info
[zed_wrapper-2] 1654705871.326778749 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/stereo/image_rect_color
[zed_wrapper-2] 1654705871.327405177 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/stereo_raw/image_raw_color
[zed_wrapper-2] 1654705871.327917679 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/confidence/confidence_map
[zed_wrapper-2] 1654705871.329223720 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/disparity/disparity_image
[zed_wrapper-2] 1654705871.329844931 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/point_cloud/cloud_registered
[zed_wrapper-2] 1654705871.330609252 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/pose
[zed_wrapper-2] 1654705871.331261696 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/pose_with_covariance
[zed_wrapper-2] 1654705871.332426803 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/odom
[zed_wrapper-2] 1654705871.333049646 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/path_map
[zed_wrapper-2] 1654705871.333632552 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/path_odom
[zed_wrapper-2] 1654705871.334475788 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/imu/data
[zed_wrapper-2] 1654705871.335021764 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/imu/data_raw
[zed_wrapper-2] 1654705871.335754948 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/temperature/imu
[zed_wrapper-2] 1654705871.336258906 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/imu/mag
[zed_wrapper-2] 1654705871.336894837 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/atm_press
[zed_wrapper-2] 1654705871.337346729 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/temperature/left
[zed_wrapper-2] 1654705871.337807421 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/temperature/right
[zed_wrapper-2] 1654705871.338554142 [zed2.zed_node] [INFO] Advertised on topic: /zed2/zed_node/left_cam_imu_transform [LATCHED]
[zed_wrapper-2] 1654705871.338606400 [zed2.zed_node] [INFO] Camera-IMU Translation: 
[zed_wrapper-2]  0 0 0
[zed_wrapper-2] 1654705871.338700484 [zed2.zed_node] [INFO] Camera-IMU Rotation: 
[zed_wrapper-2]  7FD3B8C568
[zed_wrapper-2] 1.000000 0.000000 0.000000
[zed_wrapper-2] 0.000000 1.000000 0.000000
[zed_wrapper-2] 0.000000 0.000000 1.000000
[zed_wrapper-2] 
[zed_wrapper-2] 1654705871.338811145 [zed2.zed_node] [INFO] Published static TF: 'zed2_imu_link' -> 'zed2_left_camera_frame'
[zed_wrapper-2] 1654705871.356482921 [zed2.zed_node] [INFO] *** Starting Positional Tracking ***
[zed_wrapper-2] 1654705871.356963614 [zed2.zed_node] [INFO]  * Waiting for valid static transformations...
[zed_wrapper-2] 1654705871.357472756 [zed2.zed_node] [INFO] Static transform Sensor to Base [zed2_left_camera_frame -> base_link]
[zed_wrapper-2] 1654705871.357755424 [zed2.zed_node] [INFO]  * Translation: {-0.109,-0.060,-0.397}
[zed_wrapper-2] 1654705871.357896583 [zed2.zed_node] [INFO]  * Rotation: {0.000,-15.041,0.000}
[zed_wrapper-2] 1654705871.357989163 [zed2.zed_node] [INFO] Static transform Sensor to Camera Center [zed2_left_camera_frame -> zed2_camera_center]
[zed_wrapper-2] 1654705871.358025516 [zed2.zed_node] [INFO]  * Translation: {0.000,-0.060,0.000}
[zed_wrapper-2] 1654705871.358153106 [zed2.zed_node] [INFO]  * Rotation: {0.000,-0.000,0.000}
[zed_wrapper-2] 1654705871.358252662 [zed2.zed_node] [INFO] Static transform Camera Center to Base [zed2_camera_center -> base_link]
[zed_wrapper-2] 1654705871.358342938 [zed2.zed_node] [INFO]  * Translation: {-0.109,0.000,-0.397}
[zed_wrapper-2] 1654705871.358389276 [zed2.zed_node] [INFO]  * Rotation: {0.000,-15.041,0.000}
[zed_wrapper-2] 1654705871.458715270 [zed2.zed_node] [INFO] Initial ZED left camera pose (ZED pos. tracking): 
[zed_wrapper-2] 1654705871.459129080 [zed2.zed_node] [INFO]  * T: [0.208164,0.06,0.354663]
[zed_wrapper-2] 1654705871.459296064 [zed2.zed_node] [INFO]  * Q: [0,0.130878,0,0.991398]
[zed_wrapper-2] 1654705871.478307962 [zed2.zed_node] [INFO] Odometry aligned to last tracking pose

Which clearly starts the position tracking.

### ZED Camera model

ZED2

### Environment

```shell
OS: nvidia l4t32.7
HardWare: Nvidia Jetson Xavier AGX
ZED SDK: 3.7
ROS version: Foxy
Running in docker image
```


### Anything else?

_No response_

I have a minor update.
Switching debug mode on, gives me following log before the crash:

[zed_wrapper-2] 1654866903.598805564 [zed2.zed_node] [INFO] Published static TF: 'zed2_imu_link' -> 'zed2_left_camera_frame'
[zed_wrapper-2] 1654866903.609808227 [zed2.zed_node] [DEBUG] Grab thread started
[zed_wrapper-2] 1654866903.615226804 [zed2.zed_node] [DEBUG] Sensors callback called
[zed_wrapper-2] 1654866903.615446013 [zed2.zed_node] [DEBUG] Sensors callback: counting subscribers
[zed_wrapper-2] 1654866903.616069817 [zed2.zed_node] [DEBUG] Sensors callback: Grab data and setup timestamps
[zed_wrapper-2] 1654866903.620841292 [zed2.zed_node] [DEBUG] Video Depth callback called
[ERROR] [zed_wrapper-2]: process has died [pid 12998, exit code -11, cmd '/code/ros2_ws/install/zed_wrapper/lib/zed_wrapper/zed_wrapper --ros-args -r __node:=zed_node -r __ns:=/zed2 --params-file /code/ros2_ws/install/zed_wrapper/share/zed_wrapper/config/common.yaml --params-file /code/ros2_ws/install/zed_wrapper/share/zed_wrapper/config/zed2.yaml --params-file /tmp/launch_params_9tbb9o12']

So it seems that the major issue might arise somewhere within the Video Depth CB.

--- a/last_changes.md
+++ b/last_changes.md
@@ -1,6 +1,11 @@
 LATEST CHANGES
 ==============
 
+2022-06-13
+----------
+- Fix "NO DEPTH" mode. By setting `depth/quality` to `0` now all the depth extraction and all the sub-modules depending on it are correctly disabled.
+- Add `debug` sub-set of parameters with new parameters `debug_mode` and `debug_sensors`
+
 2022-05-02
 ----------
 - Add Plane Detection

--- a/zed_components/src/zed_camera/include/zed_camera_component.hpp
+++ b/zed_components/src/zed_camera/include/zed_camera_component.hpp
@@ -278,6 +278,7 @@ private:
 
     // ----> Parameter variables
     bool mDebugMode = false;
+    bool mDebugSensors = false;
     int mCamId = 0;
     int mCamSerialNumber = 0;
     sl::MODEL mCamUserModel = sl::MODEL::ZED; // Default camera model

--- a/zed_components/src/zed_camera/src/zed_camera_component.cpp
+++ b/zed_components/src/zed_camera/src/zed_camera_component.cpp
@@ -5227,8 +5227,10 @@ void ZedCamera::applyDepthSettings()
         mRunParams.remove_saturated_areas = mRemoveSatAreas;
         mDynParMutex.unlock();
 
+        RCLCPP_DEBUG_ONCE(get_logger(), "Depth extraction enabled");
         mRunParams.enable_depth = true;
     } else {
+        RCLCPP_DEBUG_ONCE(get_logger(), "Depth extraction disabled");
         mRunParams.enable_depth = false;
     }
 }

--- a/zed_components/src/zed_camera/src/zed_camera_component.cpp
+++ b/zed_components/src/zed_camera/src/zed_camera_component.cpp
@@ -172,31 +172,34 @@ void ZedCamera::initServices()
   srv_prefix += get_name();*/
     std::string srv_prefix = "~/";
 
-    // ResetOdometry
-    srv_name = srv_prefix + mSrvResetOdomName;
-    mResetOdomSrv = create_service<std_srvs::srv::Trigger>(
-        srv_name, std::bind(&ZedCamera::callback_resetOdometry, this, _1, _2, _3));
-    RCLCPP_INFO(get_logger(), " * '%s'", mResetOdomSrv->get_service_name());
+    if(!mDepthDisabled)
+    {
+        // ResetOdometry
+        srv_name = srv_prefix + mSrvResetOdomName;
+        mResetOdomSrv = create_service<std_srvs::srv::Trigger>(
+            srv_name, std::bind(&ZedCamera::callback_resetOdometry, this, _1, _2, _3));
+        RCLCPP_INFO(get_logger(), " * '%s'", mResetOdomSrv->get_service_name());
 
-    srv_name = srv_prefix + mSrvResetPoseName;
-    mResetPosTrkSrv = create_service<std_srvs::srv::Trigger>(
-        srv_name,
-        std::bind(&ZedCamera::callback_resetPosTracking, this, _1, _2, _3));
-    RCLCPP_INFO(get_logger(), " * '%s'", mResetPosTrkSrv->get_service_name());
-    srv_name = srv_prefix + mSrvSetPoseName;
-    mSetPoseSrv = create_service<zed_interfaces::srv::SetPose>(
-        srv_name, std::bind(&ZedCamera::callback_setPose, this, _1, _2, _3));
-    RCLCPP_INFO(get_logger(), " * '%s'", mSetPoseSrv->get_service_name());
+        srv_name = srv_prefix + mSrvResetPoseName;
+        mResetPosTrkSrv = create_service<std_srvs::srv::Trigger>(
+            srv_name,
+            std::bind(&ZedCamera::callback_resetPosTracking, this, _1, _2, _3));
+        RCLCPP_INFO(get_logger(), " * '%s'", mResetPosTrkSrv->get_service_name());
+        srv_name = srv_prefix + mSrvSetPoseName;
+        mSetPoseSrv = create_service<zed_interfaces::srv::SetPose>(
+            srv_name, std::bind(&ZedCamera::callback_setPose, this, _1, _2, _3));
+        RCLCPP_INFO(get_logger(), " * '%s'", mSetPoseSrv->get_service_name());
 
-    srv_name = srv_prefix + mSrvEnableObjDetName;
-    mEnableObjDetSrv = create_service<std_srvs::srv::SetBool>(
-        srv_name, std::bind(&ZedCamera::callback_enableObjDet, this, _1, _2, _3));
-    RCLCPP_INFO(get_logger(), " * '%s'", mEnableObjDetSrv->get_service_name());
+        srv_name = srv_prefix + mSrvEnableObjDetName;
+        mEnableObjDetSrv = create_service<std_srvs::srv::SetBool>(
+            srv_name, std::bind(&ZedCamera::callback_enableObjDet, this, _1, _2, _3));
+        RCLCPP_INFO(get_logger(), " * '%s'", mEnableObjDetSrv->get_service_name());
 
-    srv_name = srv_prefix + mSrvEnableMappingName;
-    mEnableMappingSrv = create_service<std_srvs::srv::SetBool>(
-        srv_name, std::bind(&ZedCamera::callback_enableMapping, this, _1, _2, _3));
-    RCLCPP_INFO(get_logger(), " * '%s'", mEnableMappingSrv->get_service_name());
+        srv_name = srv_prefix + mSrvEnableMappingName;
+        mEnableMappingSrv = create_service<std_srvs::srv::SetBool>(
+            srv_name, std::bind(&ZedCamera::callback_enableMapping, this, _1, _2, _3));
+        RCLCPP_INFO(get_logger(), " * '%s'", mEnableMappingSrv->get_service_name());
+    }
 
     srv_name = srv_prefix + mSrvStartSvoRecName;
     mStartSvoRecSrv = create_service<zed_interfaces::srv::StartSvoRec>(
@@ -247,19 +250,40 @@ void ZedCamera::initParameters()
 
     // DEPTH parameters
     getDepthParams();
-
+    
     // POS. TRACKING parameters
-    getPosTrackingParams();
+    if (!mDepthDisabled)
+    {
+        getPosTrackingParams();        
+    }
+    else
+    {
+        mPosTrackingEnabled = false;
+    }
 
     // SENSORS parameters
     if (mCamUserModel != sl::MODEL::ZED)
         getSensorsParams();
 
-    getMappingParams();
+    if (!mDepthDisabled)
+    {
+        getMappingParams();
+    }
+    else
+    {
+        mMappingEnabled = false;
+    }
 
     // OD PARAMETERS
-    if (sl_tools::isObjDetAvailable(mCamUserModel))
-        getOdParams();
+    if (!mDepthDisabled)
+    {
+        if (sl_tools::isObjDetAvailable(mCamUserModel))
+            getOdParams();
+    }
+    else
+    {
+        mObjDetEnabled = false;
+    }
 
     // Dynamic parameters callback
     //set_on_parameters_set_callback(std::bind(&ZedCamera::callback_paramChange, this, _1)); // deprecated
@@ -273,7 +297,7 @@ void ZedCamera::getDebugParams()
 
     RCLCPP_INFO(get_logger(), "*** DEBUG parameters ***");
 
-    getParam("general.debug_mode", mDebugMode, mDebugMode);
+    getParam("debug.debug_mode", mDebugMode, mDebugMode);
     RCLCPP_INFO(get_logger(), " * Debug mode: %s", mDebugMode ? "TRUE" : "FALSE");
     if (mDebugMode) {
         rcutils_ret_t res = rcutils_logging_set_logger_level(
@@ -282,7 +306,7 @@ void ZedCamera::getDebugParams()
         if (res != RCUTILS_RET_OK) {
             RCLCPP_INFO(get_logger(), "Error setting DEBUG level fot logger");
         } else {
-            RCLCPP_INFO(get_logger(), "*** Debug Mode enabled ***");
+            RCLCPP_INFO(get_logger(), " + Debug Mode enabled +");
         }
     } else {
         rcutils_ret_t res = rcutils_logging_set_logger_level(
@@ -291,6 +315,14 @@ void ZedCamera::getDebugParams()
         if (res != RCUTILS_RET_OK) {
             RCLCPP_INFO(get_logger(), "Error setting INFO level for logger");
         }
+    }
+
+    getParam("debug.debug_sensors", mDebugSensors, mDebugSensors);
+    RCLCPP_INFO(get_logger(), " * Debug sensors: %s", mDebugSensors ? "TRUE" : "FALSE");
+
+    if(mDebugSensors && !mDebugMode)
+    {
+        RCLCPP_WARN(get_logger(), "To enable Sensors debug messages please set 'debug/debug_mode' to 'true'");
     }
 
     RCLCPP_DEBUG(get_logger(),
@@ -553,10 +585,12 @@ void ZedCamera::getDepthParams()
 
     if (mDepthQuality == sl::DEPTH_MODE::NONE) {
         mDepthDisabled = true;
+        mDepthStabilization = false;
         RCLCPP_INFO_STREAM(get_logger(),
             " * Depth quality: " << depth_quality << " - "
                                  << mDepthQuality
                                  << " - DEPTH DISABLED");
+                                 
     } else {
         mDepthDisabled = false;
         RCLCPP_INFO_STREAM(get_logger(),
@@ -1893,24 +1927,22 @@ void ZedCamera::setTFCoordFrameNames()
     RCLCPP_INFO_STREAM(get_logger(), " * Base\t\t\t-> " << mBaseFrameId);
     RCLCPP_INFO_STREAM(get_logger(), " * Camera\t\t\t-> " << mCameraFrameId);
     RCLCPP_INFO_STREAM(get_logger(), " * Left\t\t\t-> " << mLeftCamFrameId);
-    RCLCPP_INFO_STREAM(get_logger(),
-        " * Left Optical\t\t-> " << mLeftCamOptFrameId);
+    RCLCPP_INFO_STREAM(get_logger(), " * Left Optical\t\t-> " << mLeftCamOptFrameId);
     RCLCPP_INFO_STREAM(get_logger(), " * RGB\t\t\t-> " << mRgbFrameId);
     RCLCPP_INFO_STREAM(get_logger(), " * RGB Optical\t\t-> " << mRgbFrameId);
     RCLCPP_INFO_STREAM(get_logger(), " * Right\t\t\t-> " << mRightCamFrameId);
-    RCLCPP_INFO_STREAM(get_logger(),
-        " * Right Optical\t\t-> " << mRightCamOptFrameId);
-    RCLCPP_INFO_STREAM(get_logger(), " * Depth\t\t\t-> " << mDepthFrameId);
-    RCLCPP_INFO_STREAM(get_logger(),
-        " * Depth Optical\t\t-> " << mDepthOptFrameId);
-    RCLCPP_INFO_STREAM(get_logger(), " * Point Cloud\t\t-> " << mCloudFrameId);
-    RCLCPP_INFO_STREAM(get_logger(), " * Disparity\t\t-> " << mDisparityFrameId);
-    RCLCPP_INFO_STREAM(get_logger(),
-        " * Disparity Optical\t-> " << mDisparityOptFrameId);
-    RCLCPP_INFO_STREAM(get_logger(),
-        " * Confidence\t\t-> " << mConfidenceFrameId);
-    RCLCPP_INFO_STREAM(get_logger(),
-        " * Confidence Optical\t-> " << mConfidenceOptFrameId);
+    RCLCPP_INFO_STREAM(get_logger(), " * Right Optical\t\t-> " << mRightCamOptFrameId);
+    if(!mDepthDisabled)
+    {
+        RCLCPP_INFO_STREAM(get_logger(), " * Depth\t\t\t-> " << mDepthFrameId);
+        RCLCPP_INFO_STREAM(get_logger(), " * Depth Optical\t\t-> " << mDepthOptFrameId);
+        RCLCPP_INFO_STREAM(get_logger(), " * Point Cloud\t\t-> " << mCloudFrameId);
+        RCLCPP_INFO_STREAM(get_logger(), " * Disparity\t\t-> " << mDisparityFrameId);
+        RCLCPP_INFO_STREAM(get_logger(), " * Disparity Optical\t-> " << mDisparityOptFrameId);
+        RCLCPP_INFO_STREAM(get_logger(), " * Confidence\t\t-> " << mConfidenceFrameId);
+        RCLCPP_INFO_STREAM(get_logger(), " * Confidence Optical\t-> " << mConfidenceOptFrameId);
+    }
+
     if (mCamRealModel != sl::MODEL::ZED) {
         RCLCPP_INFO_STREAM(get_logger(), " * IMU\t\t\t-> " << mImuFrameId);
 
@@ -2253,21 +2285,21 @@ void ZedCamera::initPublishers()
             "Advertised on topic "
                 << mPubFusedCloud->get_topic_name() << " @ "
                 << mFusedPcPubRate << " Hz");
+    
+        std::string marker_topic = "plane_marker";
+        std::string plane_topic = "plane";
+        // Rviz markers publisher
+        mPubMarker = create_publisher<visualization_msgs::msg::Marker>(marker_topic, mMappingQos);
+        RCLCPP_INFO_STREAM(get_logger(), "Advertised on topic: " << mPubMarker->get_topic_name());
+        // Detected planes publisher
+        mPubPlane = create_publisher<zed_interfaces::msg::PlaneStamped>(plane_topic, mMappingQos);
+        RCLCPP_INFO_STREAM(get_logger(), "Advertised on topic: " << mPubPlane->get_topic_name());
     }
-    std::string marker_topic = "plane_marker";
-    std::string plane_topic = "plane";
-    // Rviz markers publisher
-    mPubMarker = create_publisher<visualization_msgs::msg::Marker>(marker_topic, mMappingQos);
-    RCLCPP_INFO_STREAM(
-        get_logger(), "Advertised on topic: " << mPubMarker->get_topic_name());
-    // Detected planes publisher
-    mPubPlane = create_publisher<zed_interfaces::msg::PlaneStamped>(plane_topic, mMappingQos);
-    RCLCPP_INFO_STREAM(
-        get_logger(), "Advertised on topic: " << mPubPlane->get_topic_name());
     // <---- Mapping
 
     // ----> Sensors
-    if (mCamRealModel != sl::MODEL::ZED) {
+    if (mCamRealModel != sl::MODEL::ZED) 
+    {
         mPubImu = create_publisher<sensor_msgs::msg::Imu>(imu_topic, mSensQos);
         RCLCPP_INFO_STREAM(get_logger(),
             "Advertised on topic: " << mPubImu->get_topic_name());
@@ -2278,7 +2310,8 @@ void ZedCamera::initPublishers()
         RCLCPP_INFO_STREAM(
             get_logger(), "Advertised on topic: " << mPubImuTemp->get_topic_name());
 
-        if (sl_tools::isZED2OrZED2i(mCamRealModel)) {
+        if (sl_tools::isZED2OrZED2i(mCamRealModel))
+        {
             mPubImuMag = create_publisher<sensor_msgs::msg::MagneticField>(
                 imu_mag_topic, mSensQos);
             RCLCPP_INFO_STREAM(
@@ -2315,14 +2348,17 @@ void ZedCamera::initPublishers()
             Liveliness: RMW_QOS_POLICY_LIVELINESS_AUTOMATIC
             Liveliness lease duration: 2147483651294967295 nanoseconds
     */
-    mClickedPtQos.reliability(RMW_QOS_POLICY_RELIABILITY_RELIABLE);
-    mClickedPtQos.durability(RMW_QOS_POLICY_DURABILITY_VOLATILE);
-    mClickedPtSub = create_subscription<geometry_msgs::msg::PointStamped>(
-        mClickedPtTopic, mClickedPtQos,
-        std::bind(&ZedCamera::callback_clickedPoint, this, _1));
+    if(!mDepthDisabled)
+    {
+        mClickedPtQos.reliability(RMW_QOS_POLICY_RELIABILITY_RELIABLE);
+        mClickedPtQos.durability(RMW_QOS_POLICY_DURABILITY_VOLATILE);
+        mClickedPtSub = create_subscription<geometry_msgs::msg::PointStamped>(
+            mClickedPtTopic, mClickedPtQos,
+            std::bind(&ZedCamera::callback_clickedPoint, this, _1));
 
-    RCLCPP_INFO_STREAM(
-        get_logger(), "Subscribed to topic " << mClickedPtTopic.c_str());
+        RCLCPP_INFO_STREAM(
+            get_logger(), "Subscribed to topic " << mClickedPtTopic.c_str());
+    }
     // <---- Subscribers
 }
 
@@ -3075,6 +3111,7 @@ void ZedCamera::stopObjDetect()
 
         objMsg->objects.clear();
 
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing EMPTY OBJ message " << mPubObjDet->get_topic_name()  );
         mPubObjDet->publish(std::move(objMsg));
         // <---- Send an empty message to indicate that no more objects are tracked
         // (e.g clean Rviz2)
@@ -3445,6 +3482,7 @@ void ZedCamera::publishStaticImuFrameAndTopic()
         mPubCamImuTransf = create_publisher<geometry_msgs::msg::TransformStamped>(
             cam_imu_tr_topic, transf_qos);
 
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing Camera-IMU transform " << mPubCamImuTransf->get_topic_name());
         mPubCamImuTransf->publish(*(mCameraImuTransfMgs.get()));
 
         RCLCPP_INFO_STREAM(get_logger(),
@@ -3470,6 +3508,7 @@ void ZedCamera::publishStaticImuFrameAndTopic()
     if (mStaticImuFramePublished) {
         return;
     }
+
 
     // Publish transformation
     mStaticTfBroadcaster->sendTransform(*(mCameraImuTransfMgs.get()));
@@ -3690,6 +3729,7 @@ void ZedCamera::threadFunc_zedGrab()
                 std::unique_lock<std::mutex> lock(mPcMutex, std::defer_lock);
 
                 if (lock.try_lock()) {
+                    RCLCPP_DEBUG(get_logger(),"Retrieving point cloud");
                     mZed.retrieveMeasure(
                         mMatCloud, sl::MEASURE::XYZBGRA, sl::MEM::CPU, mMatResolDepth);
 
@@ -3815,8 +3855,11 @@ ZedCamera::publishSensorsData(rclcpp::Time t)
 
     lastTs_imu = ts_imu;
 
-    RCLCPP_DEBUG_STREAM(
+    if(mDebugSensors) 
+    {
+        RCLCPP_DEBUG_STREAM(
         get_logger(), "SENSOR LAST PERIOD: " << dT << " sec @" << 1. / dT << " Hz");
+    }
 
     RCLCPP_DEBUG_ONCE(get_logger(), "Sensors callback: temperatures");
 
@@ -3832,22 +3875,22 @@ ZedCamera::publishSensorsData(rclcpp::Time t)
 
     // ----> Sensors freq for diagnostic
     if (new_imu_data) {
-        RCLCPP_DEBUG_ONCE(get_logger(), "Sensors callback: IMU FREQ");
+        RCLCPP_DEBUG_ONCE(get_logger(), "Sensors callback: IMU FREQ");         
         static sl_tools::StopWatch freqTimer;
 
         double mean = mImuPeriodMean_sec->addValue(freqTimer.toc());
         freqTimer.tic();
 
-        // RCLCPP_DEBUG_STREAM(get_logger(), "IMU MEAN freq: " << 1e6 / mean);
+        if(mDebugSensors)  RCLCPP_DEBUG_STREAM(get_logger(), "IMU MEAN freq: " << 1e6 / mean);
     }
 
     if (new_baro_data) {
-        RCLCPP_DEBUG_ONCE(get_logger(), "Sensors callback: BARO FREQ");
+        RCLCPP_DEBUG_ONCE(get_logger(), "Sensors callback: BARO FREQ");  
         static sl_tools::StopWatch freqTimer;
 
         double mean = mBaroPeriodMean_sec->addValue(freqTimer.toc());
         freqTimer.tic();
-        // RCLCPP_DEBUG_STREAM(get_logger(), "Barometer freq: " << 1e6/mean);
+        if(mDebugSensors)  RCLCPP_DEBUG_STREAM(get_logger(), "Barometer freq: " << 1e6/mean);
     }
 
     if (new_mag_data) {
@@ -3857,7 +3900,7 @@ ZedCamera::publishSensorsData(rclcpp::Time t)
         double mean = mMagPeriodMean_sec->addValue(freqTimer.toc());
         freqTimer.tic();
 
-        // RCLCPP_DEBUG_STREAM(get_logger(), "Magnetometer freq: " << 1e6/mean);
+        if(mDebugSensors)  RCLCPP_DEBUG_STREAM(get_logger(), "Magnetometer freq: " << 1e6/mean);
     }
     // <---- Sensors freq for diagnostic
 
@@ -3917,6 +3960,7 @@ ZedCamera::publishSensorsData(rclcpp::Time t)
             }
             // <---- Covariances copy
 
+            RCLCPP_DEBUG_STREAM(get_logger(), "Publishing IMU message");
             mPubImu->publish(std::move(imuMsg));
         } else {
             mImuPublishing = false;
@@ -3962,6 +4006,7 @@ ZedCamera::publishSensorsData(rclcpp::Time t)
             }
             // <---- Covariances copy
 
+            RCLCPP_DEBUG_STREAM(get_logger(), "Publishing IMU RAW message");
             mPubImuRaw->publish(std::move(imuRawMsg));
         }
 
@@ -3979,6 +4024,7 @@ ZedCamera::publishSensorsData(rclcpp::Time t)
             imuTempMsg->temperature = static_cast<double>(imu_temp);
             imuTempMsg->variance = 0.0;
 
+            RCLCPP_DEBUG_STREAM(get_logger(), "Publishing IMU TEMP message");
             mPubImuTemp->publish(std::move(imuTempMsg));
         }
     }
@@ -3994,6 +4040,7 @@ ZedCamera::publishSensorsData(rclcpp::Time t)
             pressMsg->fluid_pressure = sens_data.barometer.pressure * 1e2; // Pascal
             pressMsg->variance = 1.0585e-2;
 
+            RCLCPP_DEBUG_STREAM(get_logger(), "Publishing PRESS message");
             mPubPressure->publish(std::move(pressMsg));
         } else {
             mBaroPublishing = false;
@@ -4024,6 +4071,7 @@ ZedCamera::publishSensorsData(rclcpp::Time t)
             rightTempMsg->temperature = static_cast<double>(mTempRight);
             rightTempMsg->variance = 0.0;
 
+            RCLCPP_DEBUG_STREAM(get_logger(), "Publishing RIGHT TEMP message");
             mPubTempR->publish(std::move(rightTempMsg));
         }
     }
@@ -4049,6 +4097,7 @@ ZedCamera::publishSensorsData(rclcpp::Time t)
             magMsg->magnetic_field_covariance[7] = 0.0f;
             magMsg->magnetic_field_covariance[8] = 0.047e-6;
 
+            RCLCPP_DEBUG_STREAM(get_logger(), "Publishing MAG message");
             mPubImuMag->publish(std::move(magMsg));
         } else {
             mMagPublishing = false;
@@ -4260,6 +4309,8 @@ void ZedCamera::callback_pubVideoDepth()
     rclcpp::Time pub_ts;
     mPublishingData = publishVideoDepth(pub_ts);
 
+    //RCLCPP_DEBUG_ONCE(get_logger(), "Video Depth published");
+
     // RCLCPP_INFO_STREAM(get_logger(), "callback_pubVideoDepth - pub_ts type:" <<
     // pub_ts.get_clock_type());
 
@@ -4272,6 +4323,7 @@ void ZedCamera::callback_pubVideoDepth()
 
 void ZedCamera::callback_pubSensorsData()
 {
+
     RCLCPP_DEBUG_ONCE(get_logger(), "Sensors callback called");
 
     std::lock_guard<std::mutex> lock(mCloseZedMutex);
@@ -4331,10 +4383,13 @@ bool ZedCamera::publishVideoDepth(rclcpp::Time& out_pub_ts)
         rightGrayRawSubnumber = mPubRawRightGray.getNumSubscribers();
         stereoSubnumber = mPubStereo.getNumSubscribers();
         stereoRawSubnumber = mPubRawStereo.getNumSubscribers();
-        depthSubnumber = mPubDepth.getNumSubscribers();
-        depthInfoSubnumber = count_subscribers(mPubDepthInfo->get_topic_name());
-        confMapSubnumber = count_subscribers(mPubConfMap->get_topic_name());
-        disparitySubnumber = count_subscribers(mPubDisparity->get_topic_name());
+        if(!mDepthDisabled)
+        {
+            depthSubnumber = mPubDepth.getNumSubscribers();
+            depthInfoSubnumber = count_subscribers(mPubDepthInfo->get_topic_name());
+            confMapSubnumber = count_subscribers(mPubConfMap->get_topic_name());
+            disparitySubnumber = count_subscribers(mPubDisparity->get_topic_name());
+        }
     } catch (...) {
         rcutils_reset_error();
         RCLCPP_DEBUG(get_logger(),
@@ -4360,6 +4415,7 @@ bool ZedCamera::publishVideoDepth(rclcpp::Time& out_pub_ts)
     std::unique_lock<std::timed_mutex> lock(mCamDataMutex, std::defer_lock);
 
     if (lock.try_lock_for(std::chrono::milliseconds(1000 / mCamGrabFrameRate))) {
+        //RCLCPP_DEBUG(get_logger(), "Retrieving Video Data");
         if (rgbSubnumber + leftSubnumber + stereoSubnumber > 0) {
             retrieved |= sl::ERROR_CODE::SUCCESS == mZed.retrieveImage(mat_left, sl::VIEW::LEFT, sl::MEM::CPU, mMatResolVideo);
             ts_rgb = mat_left.timestamp;
@@ -4393,7 +4449,10 @@ bool ZedCamera::publishVideoDepth(rclcpp::Time& out_pub_ts)
             retrieved |= sl::ERROR_CODE::SUCCESS == mZed.retrieveImage(mat_right_raw_gray, sl::VIEW::RIGHT_UNRECTIFIED_GRAY, sl::MEM::CPU, mMatResolVideo);
             grab_ts = mat_right_raw_gray.timestamp;
         }
+        //RCLCPP_DEBUG(get_logger(), "Video Data retrieved");
+        //RCLCPP_DEBUG(get_logger(), "Retrieving Depth Data");
         if (depthSubnumber > 0 || depthInfoSubnumber > 0) {
+            RCLCPP_DEBUG(get_logger(), "Retrieving Depth");
             retrieved |= sl::ERROR_CODE::SUCCESS == mZed.retrieveMeasure(mat_depth, sl::MEASURE::DEPTH, sl::MEM::CPU, mMatResolDepth);
             grab_ts = mat_depth.timestamp;
 
@@ -4407,10 +4466,12 @@ bool ZedCamera::publishVideoDepth(rclcpp::Time& out_pub_ts)
             }
         }
         if (disparitySubnumber > 0) {
+            RCLCPP_DEBUG(get_logger(), "Retrieving Disparity");
             retrieved |= sl::ERROR_CODE::SUCCESS == mZed.retrieveMeasure(mat_disp, sl::MEASURE::DISPARITY, sl::MEM::CPU, mMatResolDepth);
             grab_ts = mat_disp.timestamp;
         }
         if (confMapSubnumber > 0) {
+            RCLCPP_DEBUG(get_logger(), "Retrieving Confidence");
             retrieved |= sl::ERROR_CODE::SUCCESS == mZed.retrieveMeasure(mat_conf, sl::MEASURE::CONFIDENCE, sl::MEM::CPU, mMatResolDepth);
             grab_ts = mat_conf.timestamp;
         }
@@ -4421,6 +4482,7 @@ bool ZedCamera::publishVideoDepth(rclcpp::Time& out_pub_ts)
             max_depth *= 0.001f;
 #endif
         }
+        //RCLCPP_DEBUG(get_logger(), "Depth Data retrieved");
     } else {
         RCLCPP_DEBUG(get_logger(), "Lock timeout");
     }
@@ -4571,6 +4633,7 @@ bool ZedCamera::publishVideoDepth(rclcpp::Time& out_pub_ts)
     // ----> Publish the side-by-side image if someone has subscribed to
     if (stereoSubnumber > 0) {
         auto combined = sl_tools::imagesToROSmsg(mat_left, mat_right, mCameraFrameId, timeStamp);
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing SIDE-BY-SIDE message");
         mPubStereo.publish(combined);
     }
     // <---- Publish the side-by-side image if someone has subscribed to
@@ -4579,6 +4642,7 @@ bool ZedCamera::publishVideoDepth(rclcpp::Time& out_pub_ts)
     if (stereoRawSubnumber > 0) {
         auto combined = sl_tools::imagesToROSmsg(
             mat_left_raw, mat_right_raw, mCameraFrameId, timeStamp);
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing SIDE-BY-SIDE RAW message");
         mPubRawStereo.publish(combined);
     }
     // <---- Publish the side-by-side image if someone has subscribed to
@@ -4591,6 +4655,7 @@ bool ZedCamera::publishVideoDepth(rclcpp::Time& out_pub_ts)
 
     // ---->  Publish the confidence image and map if someone has subscribed to
     if (confMapSubnumber > 0) {
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing CONF MAP message");
         mPubConfMap->publish(
             *sl_tools::imageToROSmsg(mat_conf, mDepthOptFrameId, timeStamp));
     }
@@ -4610,6 +4675,7 @@ bool ZedCamera::publishVideoDepth(rclcpp::Time& out_pub_ts)
         depthInfoMsg->min_depth = min_depth;
         depthInfoMsg->max_depth = max_depth;
 
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing DEPTH INFO message");
         mPubDepthInfo->publish(std::move(depthInfoMsg));
     }
     // <---- Publish the depth info if someone has subscribed to
@@ -4628,6 +4694,7 @@ void ZedCamera::publishImageWithInfo(sl::Mat& img,
 {
     auto image = sl_tools::imageToROSmsg(img, imgFrameId, t);
     camInfoMsg->header.stamp = t;
+    RCLCPP_DEBUG_STREAM(get_logger(), "Publishing IMAGE message");
     pubImg.publish(image, camInfoMsg); // TODO CHECK FOR ZERO-COPY
 }
 
@@ -4749,6 +4816,7 @@ void ZedCamera::publishOdom(tf2::Transform& odom2baseTransf,
     }
 
     // Publish odometry message
+    RCLCPP_DEBUG_STREAM(get_logger(), "Publishing ODOM message");
     mPubOdom->publish(std::move(odomMsg));
 }
 
@@ -4918,6 +4986,7 @@ void ZedCamera::publishPose()
         poseNoCov->pose = pose;
 
         // Publish pose stamped message
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing POSE NO COV message");
         mPubPose->publish(std::move(poseNoCov));
     }
 
@@ -4941,6 +5010,7 @@ void ZedCamera::publishPose()
             }
 
             // Publish pose with covariance stamped message
+            RCLCPP_DEBUG_STREAM(get_logger(), "Publishing POSE COV message");
             mPubPoseCov->publish(std::move(poseCov));
         }
     }
@@ -5105,6 +5175,7 @@ void ZedCamera::processDetectedObjects(rclcpp::Time t)
         idx++;
     }
 
+    RCLCPP_DEBUG_STREAM(get_logger(), "Publishing OBJ DET message");
     mPubObjDet->publish(std::move(objMsg));
 
     // ----> Diagnostic information update
@@ -5116,6 +5187,8 @@ void ZedCamera::processDetectedObjects(rclcpp::Time t)
 
 bool ZedCamera::isDepthRequired()
 {
+    //RCLCPP_DEBUG_STREAM(get_logger(), "isDepthRequired called");
+
     if (mDepthDisabled) {
         return false;
     }
@@ -5266,6 +5339,7 @@ void ZedCamera::publishDepthMapWithInfo(sl::Mat& depth, rclcpp::Time t)
 
     if (!mOpenniDepthMode) {
         auto depth_img = sl_tools::imageToROSmsg(depth, mDepthOptFrameId, t);
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing DEPTH message");
         mPubDepth.publish(depth_img, mDepthCamInfoMsg); // TODO CHECK FOR ZERO-COPY
         return;
     }
@@ -5297,6 +5371,7 @@ void ZedCamera::publishDepthMapWithInfo(sl::Mat& depth, rclcpp::Time t)
             std::round(*(depthDataPtr++) * 1000)); // in mm, rounded
     }
 
+    RCLCPP_DEBUG_STREAM(get_logger(), "Publishing OPENNI DEPTH message");
     mPubDepth.publish(openniDepthMsg, mDepthCamInfoMsg);
 }
 
@@ -5319,6 +5394,7 @@ void ZedCamera::publishDisparity(sl::Mat disparity, rclcpp::Time t)
     disparityMsg->min_disparity = disparityMsg->f * disparityMsg->t / mZed.getInitParameters().depth_minimum_distance;
     disparityMsg->max_disparity = disparityMsg->f * disparityMsg->t / mZed.getInitParameters().depth_maximum_distance;
 
+    RCLCPP_DEBUG_STREAM(get_logger(), "Publishing DISPARITY message");
     mPubDisparity->publish(std::move(disparityMsg));
 }
 
@@ -5375,6 +5451,7 @@ void ZedCamera::publishPointCloud()
     memcpy(ptCloudPtr, (float*)cpu_cloud, ptsCount * 4 * sizeof(float));
 
     // Pointcloud publishing
+    RCLCPP_DEBUG_STREAM(get_logger(), "Publishing POINT CLOUD message");
     mPubCloud->publish(std::move(pcMsg));
 
     // Publish freq calculation
@@ -5493,6 +5570,7 @@ void ZedCamera::callback_pubFusedPc()
     prev_ts = ros_ts;
 
     // Pointcloud publishing
+    RCLCPP_DEBUG_STREAM(get_logger(), "Publishing FUSED POINT CLOUD message");
     mPubFusedCloud->publish(std::move(pointcloudFusedMsg));
 }
 
@@ -5561,6 +5639,7 @@ void ZedCamera::callback_pubPaths()
         mapPath->header.stamp = mFrameTimestamp;
         mapPath->poses = mMapPath;
 
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing MAP PATH message");
         mPubPosePath->publish(std::move(mapPath));
     }
 
@@ -5570,6 +5649,7 @@ void ZedCamera::callback_pubPaths()
         odomPath->header.stamp = mFrameTimestamp;
         odomPath->poses = mOdomPath;
 
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing ODOM PATH message");
         mPubOdomPath->publish(std::move(odomPath));
     }
 }
@@ -6243,6 +6323,7 @@ void ZedCamera::callback_clickedPoint(const geometry_msgs::msg::PointStamped::Sh
         pt_marker->color.a = 0.8;
 
         // Publish the marker
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing PT MARKER message");
         mPubMarker->publish(std::move(pt_marker));
         // ----> Publish a blue sphere in the clicked point
 
@@ -6303,6 +6384,7 @@ void ZedCamera::callback_clickedPoint(const geometry_msgs::msg::PointStamped::Sh
         }
 
         // Publish the marker
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing PLANE MARKER message");
         mPubMarker->publish(std::move(plane_marker));
         // <---- Publish the plane as green mesh
     }
@@ -6378,6 +6460,7 @@ void ZedCamera::callback_clickedPoint(const geometry_msgs::msg::PointStamped::Sh
             planeMsg->mesh.vertices[i].z = sl_mesh.vertices[i][2];
         }
 
+        RCLCPP_DEBUG_STREAM(get_logger(), "Publishing PLANE message");
         mPubPlane->publish(std::move(planeMsg));
         // <---- Publish the plane as custom message
     }

--- a/zed_components/src/zed_camera/src/zed_camera_component.cpp
+++ b/zed_components/src/zed_camera/src/zed_camera_component.cpp
@@ -362,7 +362,7 @@ void ZedCamera::getGeneralParams()
 
     getParam("general.sdk_verbose", mVerbose, mVerbose, " * SDK Verbose: ");
     getParam("general.svo_file", std::string(), mSvoFilepath, " * SVO: ");
-    if (mSvoFilepath.compare("live") == 0) // Patch for launch file not allowing empty strings as default parameters
+    if (mSvoFilepath.compare("live") == 0) // Patch for Python launch files not allowing empty strings as default parameters
     {
         mSvoFilepath = "";
     }

--- a/zed_wrapper/config/common.yaml
+++ b/zed_wrapper/config/common.yaml
@@ -6,7 +6,7 @@
 /**:
     ros__parameters:
         general:
-            svo_file: ""
+            svo_file: "" # usually overwritten by launch file
             svo_loop: false # Enable loop mode when using an SVO as input source
             svo_realtime: false # if true SVO will be played trying to respect the original framerate eventually skipping frames, otherwise every frame will be processed respecting the `pub_frame_rate` setting
             camera_timeout_sec: 5
@@ -40,7 +40,7 @@
             qos_durability: 2 # '1': TRANSIENT_LOCAL - '2': VOLATILE
 
         depth:
-            quality: 0 # '0': NONE, '1': PERFORMANCE, '2': QUALITY, '3': ULTRA - '4': NEURAL - Note: if '0' all the modules that requires depth extraction are disabled by default (Pos. Tracking, Obj. Detection, Mapping, ...)
+            quality: 1 # '0': NONE, '1': PERFORMANCE, '2': QUALITY, '3': ULTRA - '4': NEURAL - Note: if '0' all the modules that requires depth extraction are disabled by default (Pos. Tracking, Obj. Detection, Mapping, ...)
             sensing_mode: 0 # '0': STANDARD, '1': FILL
             depth_stabilization: true # Forces positional tracking to start if 'true'
             openni_depth_mode: false # 'false': 32bit float [meters], 'true': 16bit unsigned int [millimeters]

--- a/zed_wrapper/config/common.yaml
+++ b/zed_wrapper/config/common.yaml
@@ -3,90 +3,89 @@
 #
 # Note: the parameter svo_file is passed as exe argumet
 ---
-
 /**:
     ros__parameters:
         general:
-            svo_file:                   ''
-            svo_loop:                   false                               # Enable loop mode when using an SVO as input source
-            svo_realtime:               false                               # if true SVO will be played trying to respect the original framerate eventually skipping frames, otherwise every frame will be processed respecting the `pub_frame_rate` setting
-            camera_timeout_sec:         5
-            camera_max_reconnect:       5
-            camera_flip:                false
-            zed_id:                     0
-            serial_number:              0
-            resolution:                 2                                   # '0': HD2K, '1': HD1080, '2': HD720, '3': VGA
-            sdk_verbose:                true
-            grab_frame_rate:            15                                  # ZED SDK internal grabbing rate
-            pub_frame_rate:             15.0                                # [DYNAMIC] - frequency of publishing of visual images and depth images
-            gpu_id:                     -1
+            svo_file: ""
+            svo_loop: false # Enable loop mode when using an SVO as input source
+            svo_realtime: false # if true SVO will be played trying to respect the original framerate eventually skipping frames, otherwise every frame will be processed respecting the `pub_frame_rate` setting
+            camera_timeout_sec: 5
+            camera_max_reconnect: 5
+            camera_flip: false
+            zed_id: 0
+            serial_number: 0
+            resolution: 2 # '0': HD2K, '1': HD1080, '2': HD720, '3': VGA
+            sdk_verbose: true
+            grab_frame_rate: 15 # ZED SDK internal grabbing rate
+            pub_frame_rate: 15.0 # [DYNAMIC] - frequency of publishing of visual images and depth images
+            gpu_id: -1
 
         video:
-            extrinsic_in_camera_frame:  false                               # if `false` extrinsic parameter in `camera_info` will use ROS native frame (X FORWARD, Z UP) instead of the camera frame (Z FORWARD, Y DOWN) [`true` use old behavior as for version < v3.1]
-            img_downsample_factor:      1.0                                 # Resample factor for image data matrices [0.01,1.0] The SDK works with native data sizes, but publishes rescaled matrices
-            brightness:                 4                                   # [DYNAMIC]
-            contrast:                   4                                   # [DYNAMIC]
-            hue:                        0                                   # [DYNAMIC]
-            saturation:                 4                                   # [DYNAMIC]
-            sharpness:                  4                                   # [DYNAMIC]
-            gamma:                      8                                   # [DYNAMIC] - Requires SDK >=v3.1
-            auto_exposure_gain:         true                                # [DYNAMIC]
-            exposure:                   80                                  # [DYNAMIC]
-            gain:                       80                                  # [DYNAMIC]
-            auto_whitebalance:          true                                # [DYNAMIC]
-            whitebalance_temperature:   42                                  # [DYNAMIC] - [28,65] works only if `auto_whitebalance` is false
-            qos_history:                1                                   # '1': KEEP_LAST - '2': KEEP_ALL
-            qos_depth:                  1                                   # Queue size if using KEEP_LAST
-            qos_reliability:            1                                   # '1': RELIABLE - '2': BEST_EFFORT -
-            qos_durability:             2                                   # '1': TRANSIENT_LOCAL - '2': VOLATILE
+            extrinsic_in_camera_frame: false # if `false` extrinsic parameter in `camera_info` will use ROS native frame (X FORWARD, Z UP) instead of the camera frame (Z FORWARD, Y DOWN) [`true` use old behavior as for version < v3.1]
+            img_downsample_factor: 1.0 # Resample factor for image data matrices [0.01,1.0] The SDK works with native data sizes, but publishes rescaled matrices
+            brightness: 4 # [DYNAMIC]
+            contrast: 4 # [DYNAMIC]
+            hue: 0 # [DYNAMIC]
+            saturation: 4 # [DYNAMIC]
+            sharpness: 4 # [DYNAMIC]
+            gamma: 8 # [DYNAMIC] - Requires SDK >=v3.1
+            auto_exposure_gain: true # [DYNAMIC]
+            exposure: 80 # [DYNAMIC]
+            gain: 80 # [DYNAMIC]
+            auto_whitebalance: true # [DYNAMIC]
+            whitebalance_temperature: 42 # [DYNAMIC] - [28,65] works only if `auto_whitebalance` is false
+            qos_history: 1 # '1': KEEP_LAST - '2': KEEP_ALL
+            qos_depth: 1 # Queue size if using KEEP_LAST
+            qos_reliability: 1 # '1': RELIABLE - '2': BEST_EFFORT -
+            qos_durability: 2 # '1': TRANSIENT_LOCAL - '2': VOLATILE
 
         depth:
-            quality:                    1                                   # '0': NONE, '1': PERFORMANCE, '2': QUALITY, '3': ULTRA - '4': NEURAL - Note: if '0' all the modules that requires depth extraction are disabled by default (Pos. Tracking, Obj. Detection, Mapping, ...)
-            sensing_mode:               0                                   # '0': STANDARD, '1': FILL
-            depth_stabilization:        true                                # Forces positional tracking to start if 'true'
-            openni_depth_mode:          false                               # 'false': 32bit float [meters], 'true': 16bit unsigned int [millimeters]
-            depth_downsample_factor:    1.0                                 # Resample factor for depth data matrices [0.01,1.0] The SDK works with native data sizes, but publishes rescaled matrices (depth map, point cloud, ...)
-            point_cloud_freq:           10.0                                # [DYNAMIC] - frequency of the pointcloud publishing (equal or less to `grab_frame_rate` value)
-            depth_confidence:           50                                  # [DYNAMIC]
-            depth_texture_conf:         100                                 # [DYNAMIC]
-            remove_saturated_areas:     true                                # [DYNAMIC]
-            qos_history:                1                                   # '1': KEEP_LAST - '2': KEEP_ALL
-            qos_depth:                  1                                   # Queue size if using KEEP_LAST
-            qos_reliability:            1                                   # '1': RELIABLE - '2': BEST_EFFORT -
-            qos_durability:             2                                   # '1': TRANSIENT_LOCAL - '2': VOLATILE
+            quality: 0 # '0': NONE, '1': PERFORMANCE, '2': QUALITY, '3': ULTRA - '4': NEURAL - Note: if '0' all the modules that requires depth extraction are disabled by default (Pos. Tracking, Obj. Detection, Mapping, ...)
+            sensing_mode: 0 # '0': STANDARD, '1': FILL
+            depth_stabilization: true # Forces positional tracking to start if 'true'
+            openni_depth_mode: false # 'false': 32bit float [meters], 'true': 16bit unsigned int [millimeters]
+            depth_downsample_factor: 1.0 # Resample factor for depth data matrices [0.01,1.0] The SDK works with native data sizes, but publishes rescaled matrices (depth map, point cloud, ...)
+            point_cloud_freq: 10.0 # [DYNAMIC] - frequency of the pointcloud publishing (equal or less to `grab_frame_rate` value)
+            depth_confidence: 50 # [DYNAMIC]
+            depth_texture_conf: 100 # [DYNAMIC]
+            remove_saturated_areas: true # [DYNAMIC]
+            qos_history: 1 # '1': KEEP_LAST - '2': KEEP_ALL
+            qos_depth: 1 # Queue size if using KEEP_LAST
+            qos_reliability: 1 # '1': RELIABLE - '2': BEST_EFFORT -
+            qos_durability: 2 # '1': TRANSIENT_LOCAL - '2': VOLATILE
 
         pos_tracking:
-            pos_tracking_enabled:       true                                # True to enable positional tracking from start
-            publish_tf:                 true                                # publish `odom -> base_link` TF
-            publish_map_tf:             true                                # publish `map -> odom` TF
-            base_frame:                 'base_link'                         # use the same name as in the URDF file
-            map_frame:                  'map'
-            odometry_frame:             'odom'
-            area_memory_db_path:        ''
-            area_memory:                true                                # Enable to detect loop closure
-            floor_alignment:            false                               # Enable to automatically calculate camera/floor offset
-            initial_base_pose:          [0.0,0.0,0.0, 0.0,0.0,0.0]          # Initial position of the `base_frame` -> [X, Y, Z, R, P, Y]
-            init_odom_with_first_valid_pose: true                           # Enable to initialize the odometry with the first valid pose
-            path_pub_rate:              2.0                                 # [DYNAMIC] - Camera trajectory publishing frequency
-            path_max_count:             -1                                  # use '-1' for unlimited path size
-            two_d_mode:                 false                               # Force navigation on a plane. If true the Z value will be fixed to "fixed_z_value", roll and pitch to zero
-            fixed_z_value:              0.00                                # Value to be used for Z coordinate if `two_d_mode` is true
-            qos_history:                1                                   # '1': KEEP_LAST - '2': KEEP_ALL
-            qos_depth:                  1                                   # Queue size if using KEEP_LAST
-            qos_reliability:            1                                   # '1': RELIABLE - '2': BEST_EFFORT -
-            qos_durability:             2                                   # '1': TRANSIENT_LOCAL - '2': VOLATILE
+            pos_tracking_enabled: true # True to enable positional tracking from start
+            publish_tf: true # publish `odom -> base_link` TF
+            publish_map_tf: true # publish `map -> odom` TF
+            base_frame: "base_link" # use the same name as in the URDF file
+            map_frame: "map"
+            odometry_frame: "odom"
+            area_memory_db_path: ""
+            area_memory: true # Enable to detect loop closure
+            floor_alignment: false # Enable to automatically calculate camera/floor offset
+            initial_base_pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0] # Initial position of the `base_frame` -> [X, Y, Z, R, P, Y]
+            init_odom_with_first_valid_pose: true # Enable to initialize the odometry with the first valid pose
+            path_pub_rate: 2.0 # [DYNAMIC] - Camera trajectory publishing frequency
+            path_max_count: -1 # use '-1' for unlimited path size
+            two_d_mode: false # Force navigation on a plane. If true the Z value will be fixed to "fixed_z_value", roll and pitch to zero
+            fixed_z_value: 0.00 # Value to be used for Z coordinate if `two_d_mode` is true
+            qos_history: 1 # '1': KEEP_LAST - '2': KEEP_ALL
+            qos_depth: 1 # Queue size if using KEEP_LAST
+            qos_reliability: 1 # '1': RELIABLE - '2': BEST_EFFORT -
+            qos_durability: 2 # '1': TRANSIENT_LOCAL - '2': VOLATILE
 
         mapping:
-            mapping_enabled:            false                               # True to enable mapping and fused point cloud pubblication
-            resolution:                 0.1                                 # maps resolution in meters [0.01f, 0.2f]
-            max_mapping_range:          20.0                                # maximum depth range while mapping in meters (-1 for automatic calculation) [2.0, 20.0]
-            fused_pointcloud_freq:      0.5                                 # frequency of the publishing of the fused colored point cloud
-            clicked_point_topic:        "/clicked_point"                     # Topic published by Rviz when a point of the cloud is clicked. Used for plane detection
-            qos_history:                1                                   # '1': KEEP_LAST - '2': KEEP_ALL
-            qos_depth:                  1                                   # Queue size if using KEEP_LAST
-            qos_reliability:            1                                   # '1': RELIABLE - '2': BEST_EFFORT -
-            qos_durability:             2                                   # '1': TRANSIENT_LOCAL - '2': VOLATILE
+            mapping_enabled: false # True to enable mapping and fused point cloud pubblication
+            resolution: 0.1 # maps resolution in meters [0.01f, 0.2f]
+            max_mapping_range: 20.0 # maximum depth range while mapping in meters (-1 for automatic calculation) [2.0, 20.0]
+            fused_pointcloud_freq: 0.5 # frequency of the publishing of the fused colored point cloud
+            clicked_point_topic: "/clicked_point" # Topic published by Rviz when a point of the cloud is clicked. Used for plane detection
+            qos_history: 1 # '1': KEEP_LAST - '2': KEEP_ALL
+            qos_depth: 1 # Queue size if using KEEP_LAST
+            qos_reliability: 1 # '1': RELIABLE - '2': BEST_EFFORT -
+            qos_durability: 2 # '1': TRANSIENT_LOCAL - '2': VOLATILE
 
         debug:
-            debug_general:              false
-            debug_sensors:              false
+            debug_mode: false
+            debug_sensors: false

--- a/zed_wrapper/config/common.yaml
+++ b/zed_wrapper/config/common.yaml
@@ -87,5 +87,5 @@
             qos_durability: 2 # '1': TRANSIENT_LOCAL - '2': VOLATILE
 
         debug:
-            debug_mode: false
+            debug_mode: true
             debug_sensors: false

--- a/zed_wrapper/config/zed2i.yaml
+++ b/zed_wrapper/config/zed2i.yaml
@@ -1,45 +1,43 @@
 # config/zed2i_yaml
 # Parameters for Stereolabs zed2i camera
 ---
-
 /**:
     ros__parameters:
-
         general:
-            camera_model:               'zed2i'
-            camera_name:                'zed2i'
+            camera_model: "zed2i"
+            camera_name: "zed2i"
 
         depth:
-            min_depth:                  0.3            # Min: 0.2, Max: 3.0 - Default 0.7 - Note: reducing this value wil require more computational power and GPU memory
-            max_depth:                  20.0            # Max: 40.0
+            min_depth: 0.3 # Min: 0.2, Max: 3.0 - Default 0.7 - Note: reducing this value wil require more computational power and GPU memory
+            max_depth: 20.0 # Max: 40.0
 
         pos_tracking:
-            imu_fusion:                 true            # enable/disable IMU fusion. When set to false, only the optical odometry will be used.
-            publish_imu_tf:             true            # enable/disable the static IMU TF broadcasting
+            imu_fusion: true # enable/disable IMU fusion. When set to false, only the optical odometry will be used.
+            publish_imu_tf: true # enable/disable the static IMU TF broadcasting
 
         sensors:
-            sensors_image_sync:         false           # Synchronize Sensors messages with latest published video/depth message
-            sensors_pub_rate:           200.            # frequency of publishing of sensors data. MAX: 400. - MIN: grab rate
-            qos_history:                1               # '1': KEEP_LAST - '2': KEEP_ALL
-            qos_depth:                  1               # Queue size if using KEEP_LAST
-            qos_reliability:            1               # '1': RELIABLE - '2': BEST_EFFORT -
-            qos_durability:             2               # '1': TRANSIENT_LOCAL - '2': VOLATILE
+            sensors_image_sync: false # Synchronize Sensors messages with latest published video/depth message
+            sensors_pub_rate: 200. # frequency of publishing of sensors data. MAX: 400. - MIN: grab rate
+            qos_history: 1 # '1': KEEP_LAST - '2': KEEP_ALL
+            qos_depth: 1 # Queue size if using KEEP_LAST
+            qos_reliability: 1 # '1': RELIABLE - '2': BEST_EFFORT -
+            qos_durability: 2 # '1': TRANSIENT_LOCAL - '2': VOLATILE
 
         object_detection:
-            od_enabled:                 false           # True to enable Object Detection [only ZED 2]
-            confidence_threshold:       50.0            # [DYNAMIC] - Minimum value of the detection confidence of an object [0,100]
-            model:                      0               # '0': MULTI_CLASS_BOX - '1': MULTI_CLASS_BOX_ACCURATE - '2': HUMAN_BODY_FAST - '3': HUMAN_BODY_ACCURATE - '4': MULTI_CLASS_BOX_MEDIUM - '5': HUMAN_BODY_MEDIUM - '6': PERSON_HEAD_BOX 
-            filtering_mode:             1               # '0': NONE - '1': NMS3D - '2': NMS3D_PER_CLASS
-            mc_people:                  true            # [DYNAMIC] - Enable/disable the detection of persons for 'MULTI_CLASS_X' models
-            mc_vehicle:                 true            # [DYNAMIC] - Enable/disable the detection of vehicles for 'MULTI_CLASS_X' models
-            mc_bag:                     true            # [DYNAMIC] - Enable/disable the detection of bags for 'MULTI_CLASS_X' models
-            mc_animal:                  true            # [DYNAMIC] - Enable/disable the detection of animals for 'MULTI_CLASS_X' models
-            mc_electronics:             true            # [DYNAMIC] - Enable/disable the detection of electronic devices for 'MULTI_CLASS_X' models
-            mc_fruit_vegetable:         true            # [DYNAMIC] - Enable/disable the detection of fruits and vegetables for 'MULTI_CLASS_X' models
-            mc_sport:                   true            # [DYNAMIC] - Enable/disable the detection of sport-related objects for 'MULTI_CLASS_X' models
-            body_fitting:               true           # Enable/disable body fitting for 'HUMAN_BODY_FAST' and 'HUMAN_BODY_ACCURATE' models
-            body_format:                1               # '0': POSE_18 - '1': POSE_34 [Only if `HUMAN_BODY_*` model is selected]
-            qos_history:                1               # '1': KEEP_LAST - '2': KEEP_ALL
-            qos_depth:                  1               # Queue size if using KEEP_LAST
-            qos_reliability:            1               # '1': RELIABLE - '2': BEST_EFFORT
-            qos_durability:             2               # '1': TRANSIENT_LOCAL - '2': VOLATILE
+            od_enabled: false # True to enable Object Detection [only ZED 2]
+            confidence_threshold: 50.0 # [DYNAMIC] - Minimum value of the detection confidence of an object [0,100]
+            model: 0 # '0': MULTI_CLASS_BOX - '1': MULTI_CLASS_BOX_ACCURATE - '2': HUMAN_BODY_FAST - '3': HUMAN_BODY_ACCURATE - '4': MULTI_CLASS_BOX_MEDIUM - '5': HUMAN_BODY_MEDIUM - '6': PERSON_HEAD_BOX
+            filtering_mode: 1 # '0': NONE - '1': NMS3D - '2': NMS3D_PER_CLASS
+            mc_people: true # [DYNAMIC] - Enable/disable the detection of persons for 'MULTI_CLASS_X' models
+            mc_vehicle: true # [DYNAMIC] - Enable/disable the detection of vehicles for 'MULTI_CLASS_X' models
+            mc_bag: true # [DYNAMIC] - Enable/disable the detection of bags for 'MULTI_CLASS_X' models
+            mc_animal: true # [DYNAMIC] - Enable/disable the detection of animals for 'MULTI_CLASS_X' models
+            mc_electronics: true # [DYNAMIC] - Enable/disable the detection of electronic devices for 'MULTI_CLASS_X' models
+            mc_fruit_vegetable: true # [DYNAMIC] - Enable/disable the detection of fruits and vegetables for 'MULTI_CLASS_X' models
+            mc_sport: true # [DYNAMIC] - Enable/disable the detection of sport-related objects for 'MULTI_CLASS_X' models
+            body_fitting: true # Enable/disable body fitting for 'HUMAN_BODY_FAST' and 'HUMAN_BODY_ACCURATE' models
+            body_format: 1 # '0': POSE_18 - '1': POSE_34 [Only if `HUMAN_BODY_*` model is selected]
+            qos_history: 1 # '1': KEEP_LAST - '2': KEEP_ALL
+            qos_depth: 1 # Queue size if using KEEP_LAST
+            qos_reliability: 1 # '1': RELIABLE - '2': BEST_EFFORT
+            qos_durability: 2 # '1': TRANSIENT_LOCAL - '2': VOLATILE

